From 1f890307fc545a61d95b886e9614a6720c4f0a41 Mon Sep 17 00:00:00 2001
From: devMiyax <smiyaxdev@gmail.com>
Date: Wed, 13 Nov 2019 07:58:10 +0900
Subject: [PATCH 01/12] use cpu sync scsp

(cherry picked from commit 058d38de9f0ac28322b4a1d39663475356ee628d)

# Conflicts:
#	yabause/src/retro_arena/main.cpp
---
 yabause/src/retro_arena/main.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/yabause/src/retro_arena/main.cpp b/yabause/src/retro_arena/main.cpp
index 3b961f32f..eba29bf38 100644
--- a/yabause/src/retro_arena/main.cpp
+++ b/yabause/src/retro_arena/main.cpp
@@ -246,9 +246,9 @@ int yabauseinit()
   yinit.scsp_sync_count_per_frame = g_scsp_sync;
   yinit.extend_backup = 1;
 #if defined(__JETSON__)  
-  yinit.scsp_main_mode = 1;
-#else
   yinit.scsp_main_mode = 0;
+#else
+  yinit.scsp_main_mode = 1;
 #endif
   yinit.rbg_resolution_mode = pre.getInt( "Rotate screen resolution" ,0);
 #if defined(__JETSON__)

From 61b5ae4ea8d5a1d06c5a985612add084a7ff5938 Mon Sep 17 00:00:00 2001
From: devMiyax <smiyaxdev@gmail.com>
Date: Sat, 16 Nov 2019 16:53:43 +0900
Subject: [PATCH 02/12] fix crash on sate load

(cherry picked from commit 12b5a52f89c7f0ffcd338d28e9d717870c440614)
---
 yabause/src/scsp.c | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/yabause/src/scsp.c b/yabause/src/scsp.c
index 01f9e173e..a6f349ca6 100644
--- a/yabause/src/scsp.c
+++ b/yabause/src/scsp.c
@@ -6660,15 +6660,6 @@ SoundLoadState (FILE *fp, int version, int size)
       yread(&check, (void *)&scsplines, sizeof(u32), 1, fp);
     }
 
-    if (version >= 3) {
-      yread(&check, (void *)&cdda_next_in, sizeof(u32), 1, fp);
-      yread(&check, (void *)&cdda_out_left, sizeof(u32), 1, fp);
-      yread(&check, (void *)&scsp_mute_flags, sizeof(u32), 1, fp);
-      yread(&check, (void *)&scspsoundlen, sizeof(u32), 1, fp);
-      yread(&check, (void *)&scsplines, sizeof(u32), 1, fp);
-    }
-
-
   return size;
 }
 

From e87e52d6f33251ceeab5dad5b16b948116c1c24d Mon Sep 17 00:00:00 2001
From: devMiyax <smiyaxdev@gmail.com>
Date: Sat, 16 Nov 2019 17:03:23 +0900
Subject: [PATCH 03/12] update about dialog

(cherry picked from commit 512da6fe928111e18c0674ca2b98e11ccbe77fe9)
---
 yabause/src/about.c                    | 741 ++++++++++++-------------
 yabause/src/retro_arena/MenuScreen.cpp |  22 +-
 2 files changed, 390 insertions(+), 373 deletions(-)

diff --git a/yabause/src/about.c b/yabause/src/about.c
index 031efad5b..1a3d1d8da 100644
--- a/yabause/src/about.c
+++ b/yabause/src/about.c
@@ -9,8 +9,8 @@ uint8_t about_png[] = {
 	0x06, 0x62, 0x4b, 0x47, 0x44, 0x00, 0xff, 0x00, 0xff, 0x00, 0xff, 0xa0, 
 	0xbd, 0xa7, 0x93, 0x00, 0x00, 0x00, 0x09, 0x70, 0x48, 0x59, 0x73, 0x00, 
 	0x00, 0x0b, 0x13, 0x00, 0x00, 0x0b, 0x13, 0x01, 0x00, 0x9a, 0x9c, 0x18, 
-	0x00, 0x00, 0x00, 0x07, 0x74, 0x49, 0x4d, 0x45, 0x07, 0xe3, 0x06, 0x1d, 
-	0x13, 0x1a, 0x1c, 0xcb, 0x7b, 0x02, 0xd3, 0x00, 0x00, 0x00, 0x1d, 0x69, 
+	0x00, 0x00, 0x00, 0x07, 0x74, 0x49, 0x4d, 0x45, 0x07, 0xe3, 0x0b, 0x10, 
+	0x04, 0x27, 0x2f, 0x0d, 0xd0, 0x82, 0xe2, 0x00, 0x00, 0x00, 0x1d, 0x69, 
 	0x54, 0x58, 0x74, 0x43, 0x6f, 0x6d, 0x6d, 0x65, 0x6e, 0x74, 0x00, 0x00, 
 	0x00, 0x00, 0x00, 0x43, 0x72, 0x65, 0x61, 0x74, 0x65, 0x64, 0x20, 0x77, 
 	0x69, 0x74, 0x68, 0x20, 0x47, 0x49, 0x4d, 0x50, 0x64, 0x2e, 0x65, 0x07, 
@@ -5483,7 +5483,7 @@ uint8_t about_png[] = {
 	0x8d, 0x8e, 0x38, 0x3a, 0x39, 0x89, 0x22, 0x6f, 0x02, 0x81, 0x40, 0x20, 
 	0xe8, 0x13, 0x21, 0x00, 0x0a, 0x04, 0x02, 0x81, 0xe0, 0xbe, 0xe1, 0xe6, 
 	0xe6, 0xce, 0x2b, 0x3f, 0xfc, 0x07, 0x4c, 0x26, 0x13, 0x6f, 0xbf, 0xf9, 
-	0x87, 0x9b, 0xac, 0xdb, 0x00, 0x00, 0x13, 0x3b, 0x49, 0x44, 0x41, 0x54, 
+	0x87, 0x9b, 0xac, 0xdb, 0x00, 0x00, 0x13, 0x33, 0x49, 0x44, 0x41, 0x54, 
 	0x06, 0x75, 0x75, 0x75, 0xa2, 0x53, 0x04, 0x02, 0xc1, 0x3d, 0xa1, 0xcb, 
 	0x17, 0x4d, 0xaf, 0xd7, 0xa3, 0x73, 0x71, 0xc1, 0xc9, 0xd9, 0x59, 0x74, 
 	0x8a, 0xe0, 0xa1, 0xbf, 0xa6, 0x85, 0xdf, 0x9f, 0x40, 0x20, 0x10, 0x08, 
@@ -5529,375 +5529,374 @@ uint8_t about_png[] = {
 	0x01, 0x50, 0x20, 0x10, 0x08, 0x04, 0x02, 0x81, 0x40, 0x20, 0x10, 0x08, 
 	0x04, 0x02, 0x81, 0x40, 0x20, 0x18, 0xc6, 0x08, 0x01, 0x50, 0x20, 0x10, 
 	0x08, 0x04, 0x02, 0x81, 0x40, 0x20, 0x10, 0x08, 0x04, 0x02, 0xc1, 0xff, 
-	0xdf, 0xde, 0xdd, 0x23, 0x39, 0x92, 0x1b, 0x61, 0x00, 0x9d, 0x89, 0xd8, 
-	0x6b, 0xc8, 0x90, 0xa1, 0xfb, 0xdf, 0x47, 0x1b, 0xa1, 0x8d, 0xd0, 0x1e, 
-	0x64, 0x64, 0xc8, 0x19, 0xa3, 0x89, 0x46, 0x01, 0x99, 0xf8, 0xc9, 0x7a, 
-	0xcf, 0x9c, 0x21, 0xab, 0x50, 0x60, 0x35, 0x48, 0x26, 0x01, 0x7c, 0x40, 
-	0x61, 0x0a, 0x80, 0x00, 0x00, 0x00, 0x00, 0x50, 0x98, 0x02, 0x20, 0x00, 
-	0x00, 0x00, 0x00, 0x14, 0xa6, 0x00, 0x08, 0x00, 0x00, 0x00, 0x00, 0x85, 
-	0x29, 0x00, 0x02, 0x00, 0x00, 0x00, 0x40, 0x61, 0x0a, 0x80, 0x00, 0x00, 
-	0x00, 0x00, 0x50, 0x98, 0x02, 0x20, 0x00, 0x00, 0x00, 0x00, 0x14, 0xa6, 
-	0x00, 0x08, 0x00, 0x00, 0x00, 0x00, 0x85, 0x29, 0x00, 0x02, 0x00, 0x00, 
-	0x00, 0x40, 0x61, 0x0a, 0x80, 0x00, 0x00, 0x00, 0x00, 0x50, 0x98, 0x02, 
-	0x20, 0x00, 0x00, 0x00, 0x00, 0x14, 0xf6, 0x87, 0x2e, 0x00, 0x78, 0x97, 
-	0x7f, 0xfd, 0xf3, 0x1f, 0x3a, 0x01, 0x3e, 0xf8, 0xf3, 0xaf, 0xbf, 0x75, 
-	0x02, 0x00, 0x3e, 0xab, 0x81, 0xcf, 0x6a, 0xe5, 0x28, 0x00, 0x1a, 0x7c, 
-	0xc1, 0xe0, 0x0b, 0x80, 0xcf, 0x71, 0xe0, 0x73, 0x1c, 0x00, 0x85, 0x59, 
-	0x02, 0x0c, 0x00, 0x00, 0x00, 0x00, 0x85, 0x29, 0x00, 0x02, 0x00, 0x00, 
-	0x00, 0x40, 0x61, 0x0a, 0x80, 0x00, 0x00, 0x00, 0x00, 0x50, 0x98, 0x02, 
-	0x20, 0x00, 0x00, 0x00, 0x00, 0x14, 0xa6, 0x00, 0x08, 0x00, 0x00, 0x00, 
-	0x00, 0x85, 0x29, 0x00, 0x02, 0x00, 0x00, 0x00, 0x40, 0x61, 0x0a, 0x80, 
-	0x00, 0x00, 0x00, 0x00, 0x50, 0x98, 0x02, 0x20, 0x00, 0x00, 0x00, 0x00, 
-	0x14, 0xa6, 0x00, 0x08, 0x00, 0x00, 0x00, 0x00, 0x85, 0x29, 0x00, 0x02, 
-	0x00, 0x00, 0x00, 0x40, 0x61, 0x7f, 0xe8, 0x02, 0x00, 0x5a, 0x7e, 0xfe, 
-	0xfc, 0x79, 0xfd, 0x35, 0xfc, 0xfa, 0xf5, 0xeb, 0x98, 0x6b, 0x6f, 0xb5, 
-	0x65, 0xf4, 0x7c, 0xa3, 0xd7, 0xe7, 0x75, 0x07, 0x00, 0x80, 0x77, 0x50, 
-	0x00, 0x04, 0x00, 0x00, 0x80, 0x8b, 0xdd, 0xf2, 0xc3, 0xdd, 0xd3, 0x1f, 
-	0xe7, 0xa2, 0xae, 0xeb, 0xd3, 0x79, 0x9f, 0x1e, 0x3f, 0xbb, 0xfd, 0x51, 
-	0x3f, 0x5e, 0x56, 0xbd, 0x1f, 0x98, 0xa3, 0x00, 0x68, 0xf0, 0xed, 0x6e, 
-	0x7f, 0xf6, 0x1f, 0x67, 0xd4, 0xe0, 0x98, 0xfd, 0x26, 0x61, 0xf0, 0x05, 
-	0x00, 0x00, 0x00, 0x6e, 0x62, 0x0f, 0x40, 0x00, 0x00, 0x00, 0x00, 0x28, 
-	0x4c, 0x01, 0x10, 0x00, 0x00, 0x00, 0x00, 0x0a, 0x53, 0x00, 0x04, 0x00, 
-	0x00, 0x00, 0x80, 0xc2, 0xec, 0x01, 0x08, 0x00, 0x00, 0x1f, 0x9c, 0xb4, 
-	0x97, 0xef, 0x57, 0xfb, 0xf5, 0x46, 0xb7, 0xaf, 0x77, 0x4f, 0xe0, 0xaf, 
-	0xce, 0xdb, 0xdb, 0xbe, 0x99, 0xc7, 0xcd, 0xb4, 0xa5, 0xf2, 0x7d, 0x00, 
-	0x00, 0xdf, 0x51, 0x00, 0x04, 0xa0, 0xc4, 0x97, 0x90, 0x8c, 0x2f, 0x67, 
-	0xa3, 0xd7, 0xd7, 0x6a, 0xcb, 0xea, 0x2f, 0x91, 0xad, 0xf3, 0x8d, 0x5e, 
-	0x5f, 0xf5, 0xd7, 0x1d, 0x00, 0x7c, 0x56, 0x3b, 0xe3, 0xfd, 0x39, 0x2a, 
-	0x75, 0xf7, 0x69, 0xd0, 0xe5, 0x2d, 0xc1, 0x92, 0x4f, 0x8f, 0x7f, 0xfb, 
-	0xfd, 0xc0, 0x1c, 0x05, 0x40, 0x83, 0xef, 0xb5, 0xe7, 0xcd, 0x4e, 0x2b, 
-	0xce, 0x8e, 0x6c, 0x37, 0xf8, 0x02, 0x00, 0x00, 0x00, 0x2b, 0xd8, 0x03, 
-	0x10, 0x00, 0x00, 0x00, 0x00, 0x0a, 0x53, 0x00, 0x04, 0x00, 0x00, 0x00, 
-	0x80, 0xc2, 0x2c, 0x01, 0x06, 0x00, 0x80, 0x07, 0x56, 0x6c, 0xe3, 0xd1, 
-	0xbb, 0x65, 0xc7, 0x4c, 0x5b, 0xbe, 0x3a, 0x47, 0x6f, 0xa0, 0xc6, 0xae, 
-	0x3e, 0x38, 0x69, 0x0f, 0xd2, 0x93, 0xee, 0x03, 0x00, 0xf8, 0x8e, 0x19, 
-	0x80, 0x00, 0x00, 0x00, 0x00, 0x50, 0x98, 0x19, 0x80, 0x00, 0x00, 0x00, 
-	0xf0, 0x22, 0xa7, 0xa5, 0xdc, 0xde, 0xd2, 0xce, 0xa8, 0xa0, 0xc8, 0xd3, 
-	0x66, 0xf7, 0xde, 0x72, 0x3f, 0x30, 0x47, 0x01, 0xd0, 0x1f, 0xdb, 0x74, 
-	0x3b, 0x3f, 0x9d, 0xf7, 0xe9, 0x60, 0x17, 0xf5, 0xf8, 0xec, 0x7e, 0x76, 
-	0x3f, 0xc0, 0xbe, 0xbf, 0xb9, 0xd5, 0xf7, 0xf5, 0x49, 0xd7, 0xf0, 0xb6, 
-	0x31, 0xcc, 0x78, 0x06, 0x00, 0x00, 0x71, 0x2c, 0x01, 0x06, 0x00, 0x00, 
-	0x00, 0x80, 0xc2, 0xcc, 0x00, 0x04, 0x00, 0x80, 0x49, 0x33, 0xb3, 0x5f, 
-	0xa3, 0x67, 0xb9, 0xce, 0x84, 0x67, 0x7c, 0xf5, 0xb8, 0xde, 0xe3, 0x45, 
-	0x9f, 0x77, 0xa6, 0x9f, 0x57, 0x84, 0xa8, 0x9c, 0x7e, 0x1f, 0x00, 0xc0, 
-	0xef, 0xcc, 0x00, 0x04, 0x00, 0x00, 0x00, 0x80, 0xc2, 0x14, 0x00, 0x01, 
-	0x00, 0x00, 0x00, 0xa0, 0x30, 0x4b, 0x80, 0x01, 0x00, 0x00, 0x80, 0x6d, 
-	0xa2, 0x82, 0x22, 0xb3, 0x45, 0x05, 0x51, 0xde, 0x92, 0x0e, 0x4c, 0x2d, 
-	0x0a, 0x80, 0x2c, 0x1f, 0xac, 0x6f, 0x19, 0x64, 0x0d, 0xca, 0x00, 0x00, 
-	0x00, 0x40, 0x05, 0x0a, 0x80, 0x00, 0xbc, 0x5a, 0xab, 0xa8, 0xdf, 0xfa, 
-	0x01, 0x22, 0xe3, 0xc7, 0x00, 0x3f, 0x30, 0x00, 0x11, 0x7a, 0x03, 0x3a, 
-	0x4e, 0x3a, 0xef, 0x8a, 0xf0, 0x91, 0xaf, 0x1e, 0x37, 0x13, 0x5c, 0x02, 
-	0x00, 0x37, 0xb1, 0x07, 0x20, 0x00, 0x00, 0x00, 0x00, 0x14, 0xa6, 0x00, 
-	0x08, 0x00, 0x00, 0x00, 0x00, 0x85, 0x29, 0x00, 0x02, 0x00, 0x00, 0x00, 
-	0x40, 0x61, 0xf6, 0x00, 0x04, 0x00, 0x00, 0x00, 0xae, 0x17, 0x15, 0xe4, 
-	0x78, 0x5a, 0xda, 0xaf, 0xbd, 0x49, 0x89, 0xa0, 0x00, 0x48, 0xda, 0x20, 
-	0x55, 0xb5, 0xfd, 0x06, 0x5f, 0x00, 0xa0, 0xaa, 0x5d, 0xa1, 0x18, 0xbd, 
-	0x9f, 0xd3, 0x66, 0x3e, 0xcf, 0xf9, 0x0c, 0x07, 0xc0, 0x9b, 0x59, 0x02, 
-	0x0c, 0x00, 0x00, 0x00, 0x00, 0x85, 0x99, 0x01, 0x08, 0x00, 0xc1, 0x5a, 
-	0xb3, 0x4c, 0x46, 0x67, 0xaf, 0x98, 0xb9, 0x02, 0x00, 0x00, 0x8c, 0x32, 
-	0x03, 0x10, 0x00, 0x00, 0x00, 0x00, 0x0a, 0x53, 0x00, 0x04, 0x00, 0x00, 
-	0x00, 0x80, 0xc2, 0x2c, 0x01, 0x06, 0x00, 0x80, 0x49, 0x27, 0x2d, 0xd3, 
-	0xef, 0x0d, 0xf2, 0x98, 0x09, 0xfc, 0xe8, 0x7d, 0xee, 0xae, 0xf3, 0xf6, 
-	0x3e, 0x37, 0xfa, 0xb5, 0xb4, 0x5d, 0x03, 0xc6, 0xac, 0xd8, 0xf6, 0x64, 
-	0x07, 0x39, 0x7e, 0x3a, 0xfe, 0xd3, 0x54, 0xdf, 0xec, 0xfe, 0x3f, 0xad, 
-	0x3d, 0xdc, 0x49, 0x01, 0xd0, 0xe0, 0x9b, 0x36, 0xc8, 0xee, 0xba, 0xde, 
-	0xa8, 0x08, 0xf6, 0xec, 0x41, 0xd6, 0xe0, 0x0b, 0x00, 0x00, 0x00, 0xac, 
-	0x60, 0x09, 0x30, 0x00, 0x00, 0x00, 0x00, 0x14, 0xa6, 0x00, 0x08, 0x00, 
-	0x00, 0x00, 0x00, 0x85, 0x59, 0x02, 0x0c, 0x40, 0x8a, 0x5b, 0x96, 0xb3, 
-	0xb7, 0xda, 0x99, 0xb1, 0x25, 0xc2, 0xe8, 0xf9, 0x5a, 0xff, 0xd7, 0x3a, 
-	0xe6, 0xea, 0xd7, 0xc1, 0x36, 0x06, 0x00, 0x00, 0x70, 0x1e, 0x05, 0x40, 
-	0x00, 0x00, 0x78, 0x20, 0x7b, 0xbf, 0xe4, 0xd9, 0xb6, 0xcc, 0x04, 0x6f, 
-	0xcc, 0x84, 0x6c, 0xac, 0x68, 0xdf, 0x8a, 0xf3, 0xde, 0x78, 0x1f, 0x00, 
-	0xc0, 0x77, 0x14, 0x00, 0x01, 0x00, 0x00, 0xa0, 0xa0, 0x5b, 0x0a, 0xd5, 
-	0x51, 0x41, 0x8e, 0xbb, 0x8e, 0x9f, 0x7d, 0xbd, 0x6f, 0xbb, 0x1f, 0xc8, 
-	0xa1, 0x00, 0x68, 0xf0, 0xed, 0x76, 0xda, 0xe0, 0xbb, 0x2b, 0x8d, 0x37, 
-	0xaa, 0x3d, 0x06, 0x5f, 0x00, 0x00, 0x00, 0x60, 0x05, 0x21, 0x20, 0x00, 
-	0x00, 0x00, 0x00, 0x50, 0x98, 0x02, 0x20, 0x00, 0x00, 0x00, 0x00, 0x14, 
-	0x66, 0x09, 0x30, 0x00, 0x00, 0x7c, 0x70, 0x63, 0xb2, 0x75, 0x95, 0x20, 
-	0x8f, 0x93, 0x5e, 0x5f, 0x09, 0xe7, 0x00, 0xdc, 0x4e, 0x01, 0x10, 0x80, 
-	0xd0, 0x2f, 0x85, 0x95, 0xae, 0xa1, 0xf5, 0x85, 0x6f, 0xf4, 0xda, 0x6f, 
-	0x39, 0x5f, 0xf5, 0xd7, 0x1d, 0x00, 0x00, 0xde, 0x44, 0x01, 0x10, 0x00, 
-	0x00, 0x00, 0x2e, 0x76, 0xfb, 0x2c, 0xd5, 0x5d, 0x81, 0x90, 0x51, 0xc7, 
-	0xcf, 0x7e, 0xfc, 0xdb, 0xee, 0x07, 0x72, 0x28, 0x00, 0xfa, 0x63, 0x3b, 
-	0x66, 0x30, 0xdd, 0xf5, 0x78, 0xf7, 0x03, 0x00, 0x00, 0x00, 0x50, 0x99, 
-	0x10, 0x10, 0x00, 0x00, 0x00, 0x00, 0x28, 0xcc, 0x0c, 0x40, 0x00, 0x00, 
-	0xf8, 0xa0, 0x77, 0x05, 0x41, 0x6f, 0x28, 0xc6, 0xcc, 0x73, 0xbf, 0x7a, 
-	0xdc, 0xcc, 0x79, 0x7b, 0x45, 0x9f, 0xa3, 0xf7, 0xb9, 0x27, 0xed, 0x37, 
-	0xba, 0x2b, 0xe0, 0xe4, 0xc6, 0x60, 0x15, 0x00, 0xce, 0x64, 0x06, 0x20, 
-	0x00, 0x00, 0x00, 0x00, 0x14, 0xa6, 0x00, 0x08, 0x00, 0x00, 0x00, 0x00, 
-	0x85, 0x59, 0x02, 0x0c, 0x40, 0x53, 0xf5, 0xa5, 0x46, 0xad, 0xeb, 0x6b, 
-	0x2d, 0x3f, 0x3b, 0xa9, 0x5f, 0x32, 0x96, 0xc9, 0x59, 0x62, 0x06, 0x00, 
-	0xf7, 0x78, 0xfa, 0x59, 0x20, 0x3b, 0xf8, 0x31, 0xfb, 0x73, 0x47, 0x76, 
-	0x7a, 0x6f, 0x54, 0x7f, 0x66, 0x07, 0x5d, 0xde, 0x7e, 0xff, 0xb0, 0x96, 
-	0x02, 0xe0, 0x85, 0x7f, 0x3c, 0xbb, 0x06, 0xd9, 0xa8, 0xe3, 0x9f, 0xd6, 
-	0xfe, 0x5d, 0xa9, 0xc1, 0xa7, 0xa5, 0x18, 0x1b, 0xc4, 0x01, 0x00, 0x00, 
-	0xa0, 0x26, 0x05, 0x40, 0x00, 0x00, 0x78, 0xa0, 0xf7, 0x47, 0xb3, 0x99, 
-	0x1f, 0xd7, 0x56, 0x04, 0x83, 0xf4, 0x3e, 0x77, 0xa6, 0x7d, 0x2b, 0xae, 
-	0x77, 0x26, 0x54, 0x64, 0xc5, 0xf5, 0x46, 0x3f, 0x17, 0x00, 0x46, 0xd8, 
-	0x03, 0x10, 0x00, 0x00, 0x00, 0x00, 0x0a, 0x53, 0x00, 0x04, 0x00, 0x00, 
-	0x00, 0x80, 0xc2, 0x14, 0x00, 0x01, 0x00, 0x00, 0x00, 0xa0, 0x30, 0x7b, 
-	0x00, 0x02, 0x00, 0x00, 0xc0, 0xc5, 0x9e, 0x06, 0x09, 0x46, 0x05, 0x57, 
-	0x66, 0xa7, 0x09, 0x67, 0xa7, 0xf4, 0xee, 0x7a, 0x5d, 0xa2, 0xec, 0xea, 
-	0x37, 0xa9, 0xc1, 0x77, 0x52, 0x00, 0x3c, 0x78, 0x50, 0x8e, 0x3a, 0x4e, 
-	0xf6, 0x79, 0xb3, 0xa3, 0xd6, 0xb3, 0xd3, 0x6f, 0xa3, 0x06, 0xa9, 0xa8, 
-	0x76, 0xee, 0xba, 0x5e, 0x88, 0xbe, 0x27, 0x33, 0xde, 0xe8, 0x33, 0xfe, 
-	0x3e, 0x46, 0xdb, 0xd9, 0x6a, 0x4b, 0xc6, 0x31, 0x6f, 0xf9, 0x50, 0x55, 
-	0xe1, 0x1a, 0xe0, 0xbb, 0x7b, 0x33, 0x3a, 0xc0, 0x61, 0x66, 0x6c, 0x3b, 
-	0xa9, 0x2d, 0xd1, 0xc7, 0x5b, 0x11, 0x0c, 0xd2, 0xdb, 0xe6, 0x99, 0xb0, 
-	0x95, 0x99, 0x6b, 0x9b, 0x09, 0x2e, 0x01, 0x80, 0xdf, 0x59, 0x02, 0x0c, 
-	0x00, 0x00, 0x00, 0x00, 0x85, 0x29, 0x00, 0x02, 0x00, 0x00, 0x00, 0x40, 
-	0x61, 0x0a, 0x80, 0x00, 0x00, 0x00, 0x00, 0x50, 0x98, 0x02, 0x20, 0x00, 
-	0x00, 0x00, 0x00, 0x14, 0x26, 0x04, 0x04, 0x00, 0x00, 0x1e, 0x88, 0x0e, 
-	0x61, 0x98, 0x09, 0xe8, 0xd8, 0xd5, 0x96, 0x19, 0x2b, 0x02, 0xcf, 0xa2, 
-	0x43, 0x4f, 0xa2, 0xc3, 0x3d, 0x76, 0xf5, 0x3d, 0x8c, 0xde, 0x87, 0xd9, 
-	0x7f, 0xb7, 0xb7, 0x04, 0x5d, 0x3e, 0x3d, 0xfe, 0xd3, 0xf6, 0xef, 0xea, 
-	0xff, 0x13, 0xdf, 0x2b, 0x88, 0xa7, 0x00, 0x78, 0xe1, 0xe0, 0x15, 0x35, 
-	0xe8, 0x64, 0x0f, 0xfa, 0x51, 0xe9, 0xc0, 0x51, 0x83, 0xe9, 0x69, 0x6f, 
-	0x5a, 0xbb, 0x06, 0x53, 0x29, 0xc3, 0x00, 0x00, 0x00, 0xf0, 0x2e, 0x0a, 
-	0x80, 0x00, 0xa4, 0xc8, 0x28, 0x36, 0x8f, 0x16, 0xc8, 0x5b, 0x6d, 0x69, 
-	0xfd, 0x5f, 0xeb, 0x7c, 0xa3, 0xd7, 0xb7, 0xfa, 0x1a, 0x56, 0xb7, 0x65, 
-	0xf5, 0xfd, 0xe2, 0x17, 0x68, 0x00, 0x00, 0xf8, 0x9e, 0x3d, 0x00, 0x01, 
-	0x00, 0x00, 0x00, 0xa0, 0x30, 0x05, 0x40, 0x00, 0x00, 0x00, 0x00, 0x28, 
-	0xcc, 0x12, 0x60, 0x00, 0x00, 0xf8, 0x60, 0x45, 0xf8, 0xc3, 0x8a, 0xf6, 
-	0xf5, 0xb6, 0x39, 0xba, 0x2d, 0x5f, 0x9d, 0x63, 0xd7, 0xd2, 0xfd, 0x99, 
-	0x10, 0x95, 0x15, 0xfd, 0x2c, 0x2c, 0x04, 0x80, 0x4c, 0x0a, 0x80, 0x00, 
-	0x00, 0x00, 0x50, 0x50, 0x76, 0x50, 0x61, 0x76, 0xd0, 0xe5, 0xae, 0xa0, 
-	0xc5, 0xec, 0xb4, 0xdf, 0xa8, 0x1f, 0x88, 0xb2, 0x8f, 0xff, 0xf4, 0xba, 
-	0x38, 0x9b, 0x02, 0xe0, 0x01, 0x6e, 0x1f, 0xec, 0x76, 0xa5, 0xf7, 0xee, 
-	0x8a, 0x66, 0xcf, 0x1e, 0x04, 0x77, 0xa5, 0x36, 0x03, 0x00, 0x00, 0x00, 
-	0x35, 0xd9, 0x03, 0x10, 0x00, 0x00, 0x00, 0x00, 0x0a, 0x33, 0x03, 0x10, 
-	0x80, 0xa6, 0xd6, 0xec, 0xd6, 0x8c, 0x99, 0xa6, 0x19, 0xe7, 0x1b, 0x3d, 
-	0x66, 0xeb, 0xff, 0x56, 0xb7, 0x73, 0xf4, 0x7c, 0x19, 0x4b, 0x34, 0x46, 
-	0x8f, 0x99, 0xd1, 0x9f, 0xab, 0xaf, 0x1d, 0x00, 0x00, 0x6e, 0xa4, 0x00, 
-	0x08, 0x00, 0x00, 0x0f, 0xcc, 0x14, 0x97, 0x7b, 0xc3, 0x42, 0x56, 0x04, 
-	0x74, 0xcc, 0x3c, 0x2e, 0xfa, 0xbc, 0xbd, 0x7d, 0xb5, 0xa2, 0xb0, 0xdf, 
-	0xfb, 0x7a, 0x44, 0xf7, 0x95, 0x1f, 0x2d, 0x00, 0xc8, 0x64, 0x09, 0x30, 
-	0x00, 0x00, 0x00, 0x00, 0x14, 0x66, 0x06, 0x20, 0x00, 0x00, 0x00, 0x5c, 
-	0xec, 0x69, 0x50, 0x61, 0xf6, 0x8c, 0xd3, 0xec, 0x14, 0xda, 0x4f, 0x76, 
-	0x05, 0x60, 0x9e, 0xd6, 0xce, 0xa7, 0x76, 0xdd, 0x0f, 0x66, 0x3e, 0xaf, 
-	0xa5, 0x00, 0x78, 0xc0, 0x60, 0x97, 0x9d, 0x36, 0x1b, 0x75, 0x9c, 0x5d, 
-	0x69, 0xbf, 0xd9, 0x76, 0xa5, 0x09, 0xef, 0x62, 0x90, 0x05, 0x00, 0x00, 
-	0x80, 0x77, 0xb1, 0x04, 0x18, 0x00, 0x00, 0x00, 0x00, 0x0a, 0x33, 0x03, 
-	0x10, 0x00, 0x00, 0x1e, 0xe8, 0x5d, 0x25, 0x30, 0x33, 0xeb, 0xfe, 0xa4, 
-	0x95, 0x11, 0x33, 0xa1, 0x18, 0xd1, 0x8f, 0x8b, 0xbe, 0xb6, 0x15, 0x8f, 
-	0x03, 0x80, 0x13, 0x28, 0x00, 0x02, 0xb0, 0xec, 0x4b, 0x64, 0xe6, 0x17, 
-	0xdb, 0x8c, 0x2f, 0x62, 0x19, 0xd7, 0x30, 0xda, 0xce, 0xd6, 0xf9, 0x56, 
-	0xf7, 0x75, 0x85, 0xd7, 0x01, 0x00, 0x00, 0xde, 0xc4, 0x12, 0x60, 0x00, 
-	0x00, 0x00, 0x00, 0x28, 0xcc, 0x0c, 0x40, 0x00, 0x00, 0x00, 0xb8, 0xd8, 
-	0xae, 0x60, 0xc9, 0xa8, 0xf6, 0x64, 0x1f, 0x67, 0x57, 0x2a, 0xf1, 0x2d, 
-	0x2b, 0x15, 0x76, 0xdd, 0x27, 0xac, 0xa5, 0x00, 0x78, 0xf0, 0xa0, 0x7c, 
-	0xda, 0x9b, 0xc1, 0x2d, 0xe9, 0xc6, 0xbb, 0xda, 0x73, 0xcb, 0x20, 0x28, 
-	0x82, 0x1d, 0x00, 0x00, 0x00, 0xde, 0x45, 0x01, 0x10, 0x00, 0x00, 0x26, 
-	0xcd, 0x84, 0x5d, 0xf4, 0xfe, 0x08, 0xb7, 0x2b, 0x7c, 0x64, 0x26, 0x14, 
-	0xe3, 0xa4, 0xe3, 0xed, 0x0a, 0x29, 0x59, 0x11, 0x98, 0x02, 0x00, 0xdf, 
-	0xb1, 0x07, 0x20, 0x00, 0x00, 0x00, 0x00, 0x14, 0xa6, 0x00, 0x08, 0x00, 
-	0x00, 0x00, 0x00, 0x85, 0x59, 0x02, 0x0c, 0x40, 0x53, 0xc6, 0x3e, 0x96, 
-	0xa3, 0xcb, 0x97, 0x5a, 0xcf, 0x6b, 0xb5, 0xb3, 0xf5, 0xbc, 0xd1, 0x63, 
-	0x66, 0xb4, 0xb3, 0xc2, 0x31, 0x33, 0x5e, 0x77, 0x00, 0x00, 0x60, 0x8e, 
-	0x02, 0x20, 0x00, 0x00, 0x00, 0x5c, 0x6c, 0x57, 0xca, 0x6d, 0xd4, 0x71, 
-	0xb2, 0x03, 0x33, 0x4f, 0x0b, 0xe4, 0x8c, 0xea, 0xcf, 0xd3, 0x02, 0x2d, 
-	0xfd, 0xd8, 0x7b, 0x36, 0x05, 0xc0, 0x85, 0x7f, 0xe4, 0xbb, 0x52, 0x6b, 
-	0xb3, 0x07, 0xcd, 0xa8, 0xc1, 0x62, 0xd7, 0xe0, 0x7e, 0x5a, 0xb4, 0xf9, 
-	0xae, 0xfb, 0x07, 0x00, 0x18, 0x7f, 0x8f, 0x9d, 0x79, 0x7f, 0x9d, 0x79, 
-	0x6e, 0x74, 0xe0, 0x47, 0xef, 0xe3, 0x56, 0x04, 0x56, 0x44, 0xf7, 0xe9, 
-	0xcc, 0xeb, 0x76, 0x52, 0x5b, 0x00, 0x60, 0x84, 0x3d, 0x00, 0x01, 0x00, 
-	0x00, 0x00, 0xa0, 0x30, 0x05, 0x40, 0x00, 0x00, 0x00, 0x00, 0x28, 0x4c, 
-	0x01, 0x10, 0x00, 0x00, 0x00, 0x00, 0x0a, 0x53, 0x00, 0x04, 0x00, 0x00, 
-	0x00, 0x80, 0xc2, 0x84, 0x80, 0x00, 0x30, 0xac, 0xb5, 0xb9, 0x7b, 0x6b, 
-	0x33, 0xf3, 0xd1, 0x4d, 0xe1, 0x47, 0x8f, 0x99, 0xd1, 0x96, 0x93, 0xfa, 
-	0x73, 0x54, 0x46, 0x7f, 0x66, 0xb8, 0xe5, 0xf5, 0xe3, 0x7d, 0xe3, 0xdc, 
-	0x77, 0x8f, 0xeb, 0xfd, 0x5b, 0xe9, 0x3d, 0xc7, 0x8a, 0xc0, 0x8a, 0x15, 
-	0xe1, 0x14, 0xd1, 0x61, 0x21, 0xd1, 0x63, 0x43, 0x74, 0x1f, 0x18, 0xa7, 
-	0x38, 0x71, 0x0c, 0xcb, 0x0e, 0x96, 0x7c, 0x7a, 0x9c, 0xec, 0xb4, 0xd9, 
-	0xdb, 0xd3, 0x7e, 0xa3, 0x8e, 0xb3, 0xeb, 0x7e, 0xc8, 0x4e, 0x2b, 0xa6, 
-	0x8f, 0x02, 0xe0, 0x85, 0x83, 0xf2, 0xae, 0xe3, 0x9f, 0x96, 0xd2, 0x7b, 
-	0x5a, 0xd4, 0x7d, 0xf6, 0x9b, 0xc1, 0x69, 0x6f, 0x72, 0x00, 0x00, 0x00, 
-	0xc0, 0x1d, 0x2c, 0x01, 0x06, 0x00, 0x00, 0x00, 0x80, 0xc2, 0x14, 0x00, 
-	0x01, 0x00, 0x00, 0x00, 0xa0, 0x30, 0x05, 0x40, 0x00, 0x00, 0x00, 0x00, 
-	0x28, 0xcc, 0x1e, 0x80, 0x00, 0x00, 0x90, 0xe0, 0xf4, 0x30, 0x89, 0xde, 
-	0xf6, 0xcd, 0x9c, 0xb7, 0xf7, 0x78, 0x33, 0x81, 0x29, 0x2b, 0xfa, 0x74, 
-	0x45, 0x5f, 0xad, 0x08, 0x60, 0x01, 0xe0, 0xbd, 0x14, 0x00, 0x01, 0x00, 
-	0x00, 0xe0, 0x62, 0xd9, 0xc1, 0x89, 0x51, 0xe7, 0xdd, 0xd5, 0xce, 0xec, 
-	0xf6, 0x7f, 0xf2, 0xf4, 0x38, 0xbb, 0x52, 0x71, 0x77, 0x05, 0x93, 0xb2, 
-	0x96, 0x02, 0xe0, 0x01, 0x6e, 0x4f, 0xb3, 0xdd, 0x35, 0xb8, 0x67, 0x47, 
-	0xd4, 0x47, 0x0d, 0xe2, 0xd9, 0xc7, 0x31, 0xf8, 0x72, 0xdb, 0x07, 0xd1, 
-	0x99, 0xb1, 0xac, 0x75, 0xcc, 0xd1, 0xe7, 0xdd, 0xf2, 0x7e, 0xb0, 0xfa, 
-	0xda, 0x47, 0xc7, 0xea, 0xd1, 0xe7, 0xad, 0xbe, 0x5f, 0x00, 0x00, 0xe0, 
-	0x4d, 0xec, 0x01, 0x08, 0x00, 0x00, 0x00, 0x00, 0x85, 0x29, 0x00, 0x02, 
-	0x00, 0x00, 0x00, 0x40, 0x61, 0x96, 0x00, 0x03, 0x00, 0xc0, 0x07, 0xd1, 
-	0xe1, 0x14, 0x27, 0xed, 0xef, 0x14, 0x1d, 0xd0, 0xd1, 0x7b, 0xde, 0xe8, 
-	0xd7, 0x63, 0xc5, 0x39, 0xa2, 0xdb, 0x27, 0xdc, 0x03, 0x80, 0xd5, 0xcc, 
-	0x00, 0x04, 0x00, 0x00, 0x00, 0x80, 0xc2, 0xcc, 0x00, 0x04, 0x00, 0x00, 
-	0x80, 0x8b, 0x9d, 0x96, 0xba, 0xfb, 0x74, 0x66, 0xee, 0xd3, 0x76, 0x66, 
-	0xa7, 0xeb, 0x66, 0xf7, 0xdb, 0x2d, 0xb3, 0x80, 0xa3, 0x5e, 0x97, 0xa8, 
-	0xfb, 0x84, 0x39, 0x0a, 0x80, 0x07, 0x0c, 0xbe, 0xb7, 0x0c, 0xfa, 0x51, 
-	0xd7, 0x95, 0x3d, 0x88, 0xdc, 0x72, 0x3f, 0x44, 0x3d, 0xfe, 0x96, 0xf4, 
-	0x61, 0x00, 0x00, 0x00, 0x60, 0x0f, 0x05, 0x40, 0x00, 0x52, 0xb4, 0x8a, 
-	0xd9, 0xa3, 0x05, 0xe7, 0x8c, 0x63, 0x8e, 0xca, 0x38, 0x5f, 0xeb, 0x98, 
-	0xa3, 0x3f, 0x86, 0x8c, 0x3e, 0x6f, 0xb4, 0x2d, 0xa3, 0xcf, 0xcb, 0x38, 
-	0x26, 0x00, 0x00, 0xf0, 0x7f, 0x0a, 0x80, 0x00, 0x00, 0xf0, 0x40, 0xef, 
-	0x0f, 0x00, 0x33, 0x3f, 0x14, 0x9c, 0x14, 0x26, 0x31, 0x73, 0xde, 0x93, 
-	0x42, 0x54, 0x66, 0xda, 0x32, 0xf3, 0x1a, 0x45, 0xdf, 0x1b, 0x00, 0x30, 
-	0x42, 0x08, 0x08, 0x00, 0x00, 0x00, 0x00, 0x14, 0xa6, 0x00, 0x08, 0x00, 
-	0x00, 0x00, 0x00, 0x85, 0x59, 0x02, 0x0c, 0x00, 0x00, 0x00, 0x05, 0x65, 
-	0x07, 0x39, 0x3e, 0x3d, 0x6f, 0xd4, 0xf1, 0xb3, 0xdb, 0x53, 0x55, 0x76, 
-	0xda, 0xb2, 0xfe, 0x3f, 0x9b, 0x02, 0xe0, 0xc2, 0x3f, 0xaa, 0xa8, 0xc1, 
-	0x7a, 0xd7, 0x71, 0x9e, 0x0e, 0x16, 0xbb, 0x06, 0xf1, 0x5d, 0x51, 0xf7, 
-	0xd9, 0xf7, 0xc3, 0xed, 0xfd, 0x00, 0x00, 0x00, 0x00, 0xec, 0xa1, 0x00, 
-	0x08, 0x00, 0x00, 0x0f, 0x44, 0xff, 0x98, 0xf6, 0xd5, 0x8f, 0x76, 0x33, 
-	0xa1, 0x18, 0x5f, 0x3d, 0x6e, 0xe6, 0xb9, 0xd1, 0xed, 0x5b, 0xd1, 0xa7, 
-	0x33, 0x7d, 0x3f, 0xf3, 0xdc, 0x93, 0x42, 0x4f, 0x00, 0xe0, 0x77, 0x0a, 
-	0x80, 0x00, 0x2c, 0xff, 0x52, 0xd5, 0xfa, 0x82, 0x34, 0xfa, 0xbc, 0xd1, 
-	0x2f, 0x8f, 0xa3, 0x5f, 0xae, 0xaa, 0xcf, 0xa6, 0x1d, 0xbd, 0xbe, 0x8c, 
-	0xd7, 0x36, 0xeb, 0xcb, 0x3b, 0x00, 0x00, 0xbc, 0x85, 0x10, 0x10, 0x00, 
-	0x00, 0x00, 0x00, 0x28, 0x4c, 0x01, 0x10, 0x00, 0x00, 0x00, 0x00, 0x0a, 
-	0xb3, 0x04, 0x18, 0x00, 0x00, 0x00, 0x0a, 0x8a, 0x0a, 0x06, 0x8c, 0x0a, 
-	0x78, 0xdc, 0x95, 0x36, 0xbb, 0x2b, 0x70, 0xf2, 0xb4, 0xf4, 0xe4, 0xec, 
-	0xe3, 0xec, 0x3a, 0x3e, 0x7d, 0x14, 0x00, 0x0b, 0x0d, 0xe2, 0x4f, 0x3d, 
-	0x1d, 0x7c, 0xa3, 0xda, 0x93, 0x9d, 0x5a, 0xbb, 0x6b, 0x5f, 0xae, 0xec, 
-	0xeb, 0xfa, 0x74, 0x1c, 0x69, 0xbf, 0x00, 0x70, 0xe6, 0x67, 0xb1, 0x99, 
-	0xe0, 0x8d, 0xe8, 0xcf, 0x80, 0xbb, 0x42, 0x27, 0x66, 0x42, 0x45, 0x66, 
-	0x3e, 0xdf, 0xcc, 0x5c, 0xc7, 0xcc, 0x39, 0x76, 0xf5, 0x1f, 0x00, 0x7c, 
-	0xc7, 0x12, 0x60, 0x00, 0x00, 0x00, 0x00, 0x28, 0x4c, 0x01, 0x10, 0x00, 
-	0x00, 0x00, 0x00, 0x0a, 0xb3, 0x04, 0x18, 0x80, 0xa6, 0x8c, 0xa5, 0x46, 
-	0xa3, 0xc7, 0xcc, 0x68, 0x4b, 0x6b, 0xa9, 0x57, 0xeb, 0x7c, 0xa3, 0xcf, 
-	0x5b, 0xdd, 0x96, 0x93, 0xfa, 0x73, 0x75, 0x5b, 0x00, 0x00, 0x80, 0xff, 
-	0x33, 0x03, 0x10, 0x00, 0x00, 0x00, 0x00, 0x0a, 0x33, 0x03, 0x10, 0x00, 
-	0x00, 0x26, 0xf5, 0xce, 0x7e, 0x9d, 0x09, 0x98, 0x98, 0x09, 0xd4, 0x98, 
-	0x69, 0xdf, 0x4c, 0x5b, 0xbe, 0x7a, 0x5c, 0x74, 0xb0, 0xc5, 0x8a, 0x80, 
-	0x13, 0xe1, 0x1e, 0x54, 0x93, 0x9d, 0xd2, 0xbb, 0x2b, 0x58, 0x72, 0x57, 
-	0xff, 0x64, 0xa7, 0x03, 0xdf, 0xd2, 0x0f, 0xbb, 0xc7, 0x56, 0xda, 0x14, 
-	0x00, 0x37, 0x7e, 0x00, 0xdc, 0x7d, 0xde, 0x5d, 0x11, 0xe6, 0xa7, 0xbd, 
-	0x49, 0xdc, 0x32, 0x28, 0x3f, 0x6d, 0xa7, 0x74, 0x60, 0x00, 0x00, 0x00, 
-	0xe0, 0xc7, 0x0f, 0x4b, 0x80, 0x01, 0x00, 0x00, 0x00, 0xa0, 0x34, 0x05, 
-	0x40, 0x00, 0x00, 0x00, 0x00, 0x28, 0x4c, 0x01, 0x10, 0x00, 0x00, 0x00, 
-	0x00, 0x0a, 0xb3, 0x07, 0x20, 0x00, 0x4d, 0x15, 0xf6, 0x8d, 0xcc, 0xd8, 
-	0x60, 0xb8, 0x75, 0xcc, 0xd1, 0x3e, 0x1b, 0x3d, 0x66, 0x46, 0x5b, 0x46, 
-	0xcf, 0x77, 0xd2, 0xfd, 0x69, 0x63, 0x69, 0x56, 0xde, 0x77, 0x27, 0x05, 
-	0x65, 0xac, 0x18, 0xb7, 0x67, 0x82, 0x46, 0xa2, 0x9f, 0x3b, 0xf3, 0xb8, 
-	0xe8, 0xf0, 0x96, 0xde, 0xb6, 0xac, 0x08, 0x2e, 0x01, 0x80, 0xdf, 0x29, 
-	0x00, 0x02, 0x00, 0x00, 0xc0, 0xc5, 0xa2, 0x8a, 0xc5, 0xbb, 0x8e, 0x13, 
-	0x15, 0x78, 0xf8, 0xf4, 0xf8, 0xd9, 0xfd, 0x10, 0xd5, 0x6f, 0xbb, 0x82, 
-	0x2b, 0xb3, 0xcf, 0xcb, 0x5a, 0x0a, 0x80, 0x07, 0x0f, 0xbe, 0xa7, 0x9d, 
-	0xf7, 0x96, 0x94, 0xe1, 0xd3, 0xae, 0x37, 0xea, 0xbc, 0x06, 0x59, 0x00, 
-	0x00, 0x00, 0x60, 0x84, 0x3d, 0x00, 0x01, 0x00, 0x00, 0x00, 0xa0, 0x30, 
-	0x05, 0x40, 0x00, 0x00, 0x00, 0x00, 0x28, 0xcc, 0x12, 0x60, 0x00, 0x00, 
-	0xf8, 0x60, 0x66, 0x0b, 0x8e, 0xe8, 0xc0, 0x8a, 0xaf, 0x44, 0x87, 0x67, 
-	0xf4, 0x9e, 0xa3, 0x72, 0x9b, 0x57, 0x5c, 0x5b, 0x85, 0x80, 0x2d, 0x00, 
-	0xee, 0x62, 0x06, 0x20, 0x00, 0x00, 0x00, 0x00, 0x14, 0x66, 0x06, 0x20, 
-	0x00, 0xc3, 0x4e, 0x0a, 0xa7, 0x59, 0x3d, 0x9b, 0x62, 0xf5, 0xf9, 0x5a, 
-	0x7d, 0x9d, 0xd1, 0x96, 0x5b, 0x82, 0x87, 0x04, 0x24, 0x01, 0xc0, 0x3d, 
-	0xb3, 0x4a, 0xb3, 0xd3, 0x81, 0x9f, 0x9e, 0xf7, 0x69, 0xfa, 0x6d, 0x54, 
-	0x3b, 0x77, 0x05, 0x5a, 0x0a, 0xae, 0x7c, 0x37, 0x05, 0x40, 0x83, 0xef, 
-	0xf4, 0x20, 0x15, 0x35, 0xa8, 0x65, 0xf7, 0xdb, 0xed, 0x83, 0xef, 0x2d, 
-	0xd1, 0xf5, 0x00, 0x00, 0x00, 0xc0, 0x59, 0x2c, 0x01, 0x06, 0x00, 0x00, 
-	0x00, 0x80, 0xc2, 0xcc, 0x00, 0x04, 0x00, 0x80, 0x0f, 0x4e, 0x5a, 0xd9, 
-	0xf1, 0xd5, 0xec, 0xfe, 0xaf, 0xfe, 0x2d, 0x3a, 0x88, 0x22, 0x3a, 0x08, 
-	0xa5, 0x37, 0xa0, 0x23, 0xba, 0x7d, 0x33, 0xaf, 0x65, 0x74, 0x10, 0x4a, 
-	0x74, 0x3f, 0x03, 0xc0, 0x77, 0xcc, 0x00, 0x04, 0x00, 0x00, 0x00, 0x80, 
-	0xc2, 0x14, 0x00, 0x01, 0x00, 0x00, 0x00, 0xa0, 0x30, 0x4b, 0x80, 0x01, 
-	0x00, 0x00, 0xa0, 0xa0, 0x5d, 0x4b, 0xcb, 0xa3, 0xb6, 0x4f, 0xc8, 0x4e, 
-	0xad, 0x7d, 0x7a, 0xde, 0x5d, 0xd7, 0xb5, 0x2b, 0x10, 0x72, 0x57, 0x6a, 
-	0x33, 0x39, 0x14, 0x00, 0x0d, 0xbe, 0xcb, 0x1f, 0x1f, 0x95, 0x32, 0xfc, 
-	0xb4, 0x3d, 0xbb, 0xf6, 0xf0, 0xd9, 0x35, 0xf8, 0x9e, 0x76, 0x1f, 0xf2, 
-	0x3e, 0x19, 0x7f, 0x73, 0x19, 0xf7, 0x6f, 0xeb, 0x98, 0xa3, 0xd7, 0x30, 
-	0x7a, 0xcc, 0xd5, 0xe3, 0xd4, 0x49, 0x7b, 0x9b, 0xb5, 0xda, 0x62, 0xdc, 
-	0x02, 0x00, 0x80, 0x39, 0x0a, 0x80, 0x00, 0x00, 0xf0, 0xc0, 0x8a, 0xa2, 
-	0xf4, 0x8d, 0x3f, 0x5c, 0x46, 0x07, 0x7e, 0xac, 0x68, 0xcb, 0x8a, 0x7e, 
-	0x39, 0xe9, 0xc7, 0x16, 0x00, 0xde, 0xcb, 0x1e, 0x80, 0x00, 0x00, 0x00, 
-	0x00, 0x50, 0x98, 0x02, 0x20, 0x00, 0x00, 0x00, 0x00, 0x14, 0xa6, 0x00, 
-	0x08, 0x00, 0x00, 0x00, 0x00, 0x85, 0xd9, 0x03, 0x10, 0x00, 0x00, 0x00, 
-	0x5e, 0xe4, 0xb4, 0x94, 0xde, 0xec, 0x54, 0xdc, 0x28, 0xd9, 0xe9, 0xc0, 
-	0xd9, 0x81, 0x9c, 0x51, 0xed, 0xf9, 0xd4, 0x0f, 0xd9, 0x81, 0x9f, 0xcc, 
-	0x51, 0x00, 0x34, 0xf8, 0x96, 0x1b, 0xac, 0xa3, 0x06, 0x9d, 0xdb, 0x07, 
-	0x5f, 0x00, 0xe0, 0x8e, 0xcf, 0x73, 0xd1, 0x01, 0x13, 0xbd, 0xc7, 0xdb, 
-	0x75, 0xde, 0x99, 0xb6, 0x44, 0x9f, 0x77, 0x57, 0x61, 0xe1, 0xa4, 0xeb, 
-	0x05, 0xe0, 0x1d, 0x14, 0x00, 0x01, 0x60, 0xa1, 0xd6, 0x17, 0xb7, 0x8c, 
-	0x2f, 0x9d, 0x19, 0xe7, 0xcb, 0xf8, 0xf2, 0x79, 0x4b, 0x3b, 0x01, 0x00, 
-	0xe0, 0x46, 0xf6, 0x00, 0x04, 0x00, 0x00, 0x00, 0x80, 0xc2, 0x14, 0x00, 
-	0x01, 0x00, 0x00, 0x00, 0xa0, 0x30, 0x05, 0x40, 0x00, 0x00, 0x00, 0x00, 
-	0x28, 0xcc, 0x1e, 0x80, 0x00, 0x00, 0x50, 0xc8, 0xae, 0x10, 0x8b, 0xde, 
-	0xc0, 0x8a, 0x99, 0xc7, 0xcd, 0x5c, 0xef, 0x8a, 0xb0, 0x90, 0xe8, 0x73, 
-	0xd8, 0xcb, 0x14, 0xe3, 0xd9, 0x59, 0x41, 0x88, 0x51, 0xed, 0x79, 0x3a, 
-	0x4e, 0xef, 0x4a, 0xd7, 0xfd, 0x74, 0x7c, 0x01, 0x95, 0x77, 0x52, 0x00, 
-	0x64, 0x7a, 0x70, 0xc9, 0x3e, 0xfe, 0x69, 0x11, 0xef, 0xb7, 0xbc, 0x39, 
-	0xdd, 0xde, 0x4e, 0x00, 0x00, 0x00, 0x20, 0x86, 0x25, 0xc0, 0x00, 0x00, 
-	0x00, 0x00, 0x50, 0x98, 0x19, 0x80, 0x00, 0xb0, 0x50, 0x6b, 0x56, 0x73, 
-	0xc6, 0x4c, 0xdc, 0xd5, 0xe7, 0x1b, 0x6d, 0x4b, 0x4b, 0xab, 0x9d, 0xbb, 
-	0x66, 0x89, 0x03, 0x00, 0xc0, 0x4d, 0xcc, 0x00, 0x04, 0x00, 0x00, 0x00, 
-	0x80, 0xc2, 0xcc, 0x00, 0x04, 0x00, 0x80, 0x4b, 0xf5, 0xce, 0xe4, 0xdd, 
-	0x15, 0x94, 0x71, 0x63, 0x1f, 0xf4, 0x9e, 0x63, 0x26, 0xb8, 0xa4, 0xf7, 
-	0x1c, 0x2b, 0x82, 0x46, 0x00, 0x78, 0x07, 0x05, 0x40, 0x00, 0x00, 0x00, 
-	0xe0, 0x38, 0xbb, 0xd2, 0x75, 0x4f, 0xbb, 0xde, 0xa7, 0xd7, 0xb5, 0x6b, 
-	0x8b, 0x14, 0x81, 0x93, 0x67, 0x53, 0x00, 0x24, 0x6d, 0x30, 0x7a, 0xfa, 
-	0x47, 0x9e, 0x9d, 0xba, 0x7b, 0xcb, 0xa0, 0x79, 0x4b, 0x7b, 0x00, 0x00, 
-	0x00, 0x80, 0x3b, 0xd8, 0x03, 0x10, 0x00, 0x00, 0x00, 0x00, 0x0a, 0x53, 
-	0x00, 0x04, 0x00, 0x00, 0x00, 0x80, 0xc2, 0x2c, 0x01, 0x06, 0x20, 0xc5, 
-	0x2d, 0x7b, 0x7d, 0xb4, 0x96, 0xd7, 0xb7, 0xae, 0x61, 0x74, 0x59, 0x7e, 
-	0x85, 0x7e, 0xa9, 0x7e, 0x4f, 0xc0, 0x4d, 0xf7, 0x77, 0x6f, 0x48, 0xc4, 
-	0x4c, 0x50, 0xc6, 0x4c, 0xd8, 0x45, 0x74, 0x88, 0x45, 0xf4, 0x75, 0xac, 
-	0xe8, 0xfb, 0x99, 0xbe, 0x02, 0x80, 0x28, 0x66, 0x00, 0x02, 0x00, 0x00, 
-	0x00, 0x40, 0x61, 0x66, 0x00, 0x02, 0x00, 0x00, 0xc0, 0x8b, 0x9c, 0x36, 
-	0x2b, 0x3f, 0x3b, 0xe0, 0x31, 0xea, 0x7a, 0xa3, 0x82, 0x31, 0xb3, 0x03, 
-	0x36, 0xb3, 0xfb, 0x33, 0xbb, 0x7f, 0xc8, 0xa1, 0x00, 0x68, 0xf0, 0x4d, 
-	0xfb, 0xe3, 0xdf, 0x35, 0x18, 0x45, 0x3d, 0x3e, 0xbb, 0x3d, 0xd9, 0x6f, 
-	0x66, 0x96, 0x91, 0x00, 0x00, 0x00, 0x00, 0x3f, 0x7e, 0x58, 0x02, 0x0c, 
-	0x00, 0x00, 0x00, 0x00, 0xa5, 0x99, 0x01, 0x08, 0x00, 0x00, 0x0f, 0x9c, 
-	0x3e, 0xcb, 0x3e, 0x3a, 0x00, 0xa3, 0xf7, 0x7a, 0x7b, 0x03, 0x30, 0x56, 
-	0x04, 0x74, 0xcc, 0x84, 0x71, 0x9c, 0xd4, 0xf7, 0x00, 0x10, 0xc5, 0x0c, 
-	0x40, 0x00, 0x00, 0x00, 0x00, 0x28, 0xcc, 0x0c, 0x40, 0x00, 0x86, 0xbd, 
-	0x79, 0xaf, 0xc9, 0xd6, 0x8c, 0x8e, 0x56, 0xbf, 0x64, 0x3c, 0x6f, 0xf5, 
-	0xf5, 0x65, 0xdc, 0x2f, 0x66, 0xc8, 0x00, 0x00, 0x40, 0x1e, 0x05, 0x40, 
-	0x00, 0x00, 0x00, 0x28, 0xe8, 0xf6, 0x1f, 0x6b, 0x77, 0xa5, 0xd9, 0xee, 
-	0x6a, 0x7f, 0x94, 0x5d, 0x29, 0xc3, 0x4f, 0xdb, 0xc3, 0x5a, 0x0a, 0x80, 
-	0x06, 0xdf, 0xb4, 0x41, 0xe4, 0xb4, 0x28, 0xf7, 0xd3, 0xa2, 0xc7, 0x77, 
-	0xdd, 0x0f, 0x06, 0x5f, 0x00, 0x00, 0x00, 0x78, 0x17, 0x05, 0x40, 0x00, 
-	0x00, 0xf8, 0xe0, 0xf4, 0x1f, 0xce, 0x56, 0x84, 0x67, 0xec, 0x3a, 0xde, 
-	0xe9, 0xd7, 0x06, 0x00, 0x37, 0x11, 0x02, 0x02, 0x00, 0x00, 0x00, 0x00, 
-	0x85, 0x29, 0x00, 0x02, 0x00, 0x00, 0x00, 0x40, 0x61, 0x0a, 0x80, 0x00, 
-	0x00, 0x00, 0x00, 0x50, 0x98, 0x3d, 0x00, 0x01, 0x68, 0xaa, 0xbe, 0x67, 
-	0x52, 0xc6, 0xf5, 0x8d, 0x1e, 0xf3, 0xa4, 0xb6, 0x54, 0xef, 0x6b, 0x00, 
-	0xf0, 0x79, 0xa6, 0x5e, 0xfb, 0x77, 0xf5, 0xc3, 0x2d, 0xed, 0xbf, 0xbd, 
-	0x9f, 0x99, 0xa3, 0x00, 0x68, 0xf0, 0x9d, 0x6e, 0xbf, 0xc1, 0xce, 0x7d, 
-	0x08, 0x00, 0x00, 0x00, 0x9c, 0xcb, 0x12, 0x60, 0x00, 0x00, 0x00, 0x00, 
-	0x28, 0x4c, 0x01, 0x10, 0x00, 0x00, 0x00, 0x00, 0x0a, 0x53, 0x00, 0x04, 
-	0x00, 0x00, 0x00, 0x80, 0xc2, 0x14, 0x00, 0x01, 0x00, 0x00, 0x00, 0xa0, 
-	0x30, 0x05, 0x40, 0x00, 0x00, 0x00, 0x00, 0x28, 0x4c, 0x01, 0x10, 0x00, 
-	0x00, 0x00, 0x00, 0x0a, 0x53, 0x00, 0x04, 0x00, 0x00, 0x00, 0x80, 0xc2, 
-	0x14, 0x00, 0x01, 0x00, 0x00, 0x00, 0xa0, 0x30, 0x05, 0x40, 0x00, 0x00, 
-	0x00, 0x00, 0x28, 0x4c, 0x01, 0x10, 0x00, 0x00, 0x00, 0x00, 0x0a, 0x53, 
-	0x00, 0x04, 0x00, 0x00, 0x00, 0x80, 0xc2, 0xfe, 0xd0, 0x05, 0x00, 0xef, 
-	0xf2, 0xe7, 0x5f, 0x7f, 0xeb, 0x04, 0x00, 0x00, 0x9f, 0xd5, 0x80, 0x17, 
-	0xf9, 0xf9, 0xef, 0xff, 0xfc, 0xf7, 0x97, 0x6e, 0x00, 0x00, 0x00, 0x00, 
-	0x80, 0x9a, 0x2c, 0x01, 0x06, 0x00, 0x00, 0x00, 0x80, 0xc2, 0x14, 0x00, 
-	0x01, 0x00, 0x00, 0x00, 0xa0, 0x30, 0x05, 0x40, 0x00, 0x00, 0x00, 0x00, 
-	0x28, 0x4c, 0x01, 0x10, 0x00, 0x00, 0x00, 0x00, 0x0a, 0x53, 0x00, 0x04, 
-	0x00, 0x00, 0x00, 0x80, 0xc2, 0x14, 0x00, 0x01, 0x00, 0x00, 0x00, 0xa0, 
-	0x30, 0x05, 0x40, 0x00, 0x00, 0x00, 0x00, 0x28, 0x4c, 0x01, 0x10, 0x00, 
-	0x00, 0x00, 0x00, 0x0a, 0x53, 0x00, 0x04, 0x00, 0x00, 0x00, 0x80, 0xc2, 
-	0x14, 0x00, 0x01, 0x00, 0x00, 0x00, 0xa0, 0x30, 0x05, 0x40, 0x00, 0x00, 
-	0x00, 0x00, 0x28, 0x4c, 0x01, 0x10, 0x00, 0x00, 0x00, 0x00, 0x0a, 0x53, 
-	0x00, 0x04, 0x00, 0x00, 0x00, 0x80, 0xc2, 0x14, 0x00, 0x01, 0x00, 0x00, 
-	0x00, 0xa0, 0x30, 0x05, 0x40, 0x00, 0x00, 0x00, 0x00, 0x28, 0x4c, 0x01, 
-	0x10, 0x00, 0x00, 0x00, 0x00, 0x0a, 0x53, 0x00, 0x04, 0x00, 0x00, 0x00, 
-	0x80, 0xc2, 0x14, 0x00, 0x01, 0x00, 0x00, 0x00, 0xa0, 0x30, 0x05, 0x40, 
-	0x00, 0x00, 0x00, 0x00, 0x28, 0x4c, 0x01, 0x10, 0x00, 0x00, 0x00, 0x00, 
-	0x0a, 0x53, 0x00, 0x04, 0x00, 0x00, 0x00, 0x80, 0xc2, 0x14, 0x00, 0x01, 
-	0x00, 0x00, 0x00, 0xa0, 0x30, 0x05, 0x40, 0x00, 0x00, 0x00, 0x00, 0x28, 
-	0x4c, 0x01, 0x10, 0x00, 0x00, 0x00, 0x00, 0x0a, 0x53, 0x00, 0x04, 0x00, 
-	0x00, 0x00, 0x80, 0xc2, 0x14, 0x00, 0x01, 0x00, 0x00, 0x00, 0xa0, 0x30, 
-	0x05, 0x40, 0x00, 0x00, 0x00, 0x00, 0x28, 0x4c, 0x01, 0x10, 0x00, 0x00, 
-	0x00, 0x00, 0x0a, 0x53, 0x00, 0x04, 0x00, 0x00, 0x00, 0x80, 0xc2, 0x14, 
-	0x00, 0x01, 0x00, 0x00, 0x00, 0xa0, 0x30, 0x05, 0x40, 0x00, 0x00, 0x00, 
-	0x00, 0x28, 0x4c, 0x01, 0x10, 0x00, 0x00, 0x00, 0x00, 0x0a, 0x53, 0x00, 
-	0x04, 0x00, 0x00, 0x00, 0x80, 0xc2, 0x14, 0x00, 0x01, 0x00, 0x00, 0x00, 
-	0xa0, 0x30, 0x05, 0x40, 0x00, 0x00, 0x00, 0x00, 0x28, 0x4c, 0x01, 0x10, 
-	0x00, 0x00, 0x00, 0x00, 0x0a, 0x53, 0x00, 0x04, 0x00, 0x00, 0x00, 0x80, 
-	0xc2, 0x14, 0x00, 0x01, 0x00, 0x00, 0x00, 0xa0, 0xb0, 0xff, 0x01, 0xa4, 
-	0x63, 0xf5, 0x5f, 0xa1, 0x8b, 0x05, 0x83, 0x00, 0x00, 0x00, 0x00, 0x49, 
-	0x45, 0x4e, 0x44, 0xae, 0x42, 0x60, 0x82, 
+	0xdf, 0xde, 0x1d, 0x23, 0xb9, 0x6e, 0x23, 0x61, 0x00, 0x9e, 0x57, 0xe5, 
+	0x6b, 0x38, 0xd8, 0x60, 0xef, 0x7f, 0x20, 0x07, 0x0e, 0x7c, 0x90, 0xb7, 
+	0x81, 0xa3, 0xad, 0x1a, 0x71, 0x48, 0xa0, 0x1b, 0x0d, 0x34, 0xbf, 0x2f, 
+	0x1d, 0x09, 0x04, 0x21, 0x0a, 0xd2, 0xb4, 0x40, 0xfc, 0x40, 0x63, 0x0a, 
+	0x80, 0x00, 0x00, 0x00, 0x00, 0xd0, 0x98, 0x02, 0x20, 0x00, 0x00, 0x00, 
+	0x00, 0x34, 0xa6, 0x00, 0x08, 0x00, 0x00, 0x00, 0x00, 0x8d, 0x29, 0x00, 
+	0x02, 0x00, 0x00, 0x00, 0x40, 0x63, 0x0a, 0x80, 0x00, 0x00, 0x00, 0x00, 
+	0xd0, 0x98, 0x02, 0x20, 0x00, 0x00, 0x00, 0x00, 0x34, 0xa6, 0x00, 0x08, 
+	0x00, 0x00, 0x00, 0x00, 0x8d, 0x29, 0x00, 0x02, 0x00, 0x00, 0x00, 0x40, 
+	0x63, 0x0a, 0x80, 0x00, 0x00, 0x00, 0x00, 0xd0, 0x98, 0x02, 0x20, 0x00, 
+	0x00, 0x00, 0x00, 0x34, 0xf6, 0x87, 0x21, 0x00, 0x78, 0x97, 0xff, 0xfe, 
+	0xe7, 0x4f, 0x83, 0x00, 0x1f, 0xfc, 0xf5, 0xf7, 0x3f, 0x06, 0xc1, 0xbc, 
+	0x07, 0xe6, 0x3d, 0xcc, 0x59, 0x60, 0xce, 0x6a, 0x47, 0x01, 0xd0, 0xe4, 
+	0x0b, 0x26, 0x5f, 0x00, 0x00, 0x00, 0x68, 0xcc, 0x2d, 0xc0, 0x00, 0x00, 
+	0x00, 0x00, 0xd0, 0x98, 0x02, 0x20, 0x00, 0x00, 0x00, 0x00, 0x34, 0xa6, 
+	0x00, 0x08, 0x00, 0x00, 0x00, 0x00, 0x8d, 0x29, 0x00, 0x02, 0x00, 0x00, 
+	0x00, 0x40, 0x63, 0x0a, 0x80, 0x00, 0x00, 0x00, 0x00, 0xd0, 0x98, 0x02, 
+	0x20, 0x00, 0x00, 0x00, 0x00, 0x34, 0xa6, 0x00, 0x08, 0x00, 0x00, 0x00, 
+	0x00, 0x8d, 0x29, 0x00, 0x02, 0x00, 0x00, 0x00, 0x40, 0x63, 0x0a, 0x80, 
+	0x00, 0x00, 0x00, 0x00, 0xd0, 0x98, 0x02, 0x20, 0x00, 0x00, 0x00, 0x00, 
+	0x34, 0xf6, 0x87, 0x21, 0x00, 0xe0, 0xca, 0xaf, 0x5f, 0xbf, 0x8e, 0x3f, 
+	0x87, 0xdf, 0xbf, 0x7f, 0x6f, 0x73, 0xee, 0x57, 0x7d, 0x19, 0x3d, 0xde, 
+	0xe8, 0xf9, 0x79, 0xdd, 0x01, 0x00, 0xe0, 0x1d, 0x14, 0x00, 0x01, 0x00, 
+	0x00, 0xe0, 0x60, 0xa7, 0xfc, 0x70, 0xf7, 0xf4, 0xc7, 0xb9, 0xa8, 0xf3, 
+	0xfa, 0x74, 0xdc, 0xa7, 0xed, 0x67, 0xf7, 0x3f, 0xea, 0xc7, 0xcb, 0xae, 
+	0xd7, 0x03, 0x73, 0x14, 0x00, 0x4d, 0xbe, 0xb7, 0xfb, 0x9f, 0xfd, 0xe6, 
+	0x8c, 0x9a, 0x1c, 0xb3, 0x3f, 0x24, 0x4c, 0xbe, 0x00, 0x00, 0x00, 0xc0, 
+	0x49, 0xec, 0x01, 0x08, 0x00, 0x00, 0x00, 0x00, 0x8d, 0x29, 0x00, 0x02, 
+	0x00, 0x00, 0x00, 0x40, 0x63, 0x6e, 0x01, 0x06, 0x00, 0x80, 0x1f, 0xec, 
+	0xb4, 0xa5, 0xc7, 0xdd, 0x6d, 0x3b, 0xbe, 0xeb, 0xf3, 0xcc, 0x96, 0x1f, 
+	0x2b, 0xda, 0x3b, 0x71, 0x4c, 0x77, 0xef, 0x33, 0x00, 0x7c, 0x7d, 0x59, 
+	0x01, 0x08, 0x00, 0x00, 0x00, 0x00, 0xad, 0x59, 0x01, 0x08, 0xc0, 0xb0, 
+	0x9d, 0x56, 0x1f, 0x64, 0xac, 0xca, 0x18, 0x3d, 0xbf, 0xab, 0xbe, 0xac, 
+	0x5e, 0x3d, 0x72, 0x75, 0xbc, 0xd1, 0xf3, 0xeb, 0xfe, 0xba, 0x03, 0x80, 
+	0xef, 0x6a, 0x7b, 0x7c, 0x3e, 0x47, 0xa5, 0xee, 0x3e, 0x0d, 0xba, 0x3c, 
+	0x25, 0x58, 0xf2, 0x69, 0xfb, 0xa7, 0x5f, 0x0f, 0xcc, 0x51, 0x00, 0x34, 
+	0xf9, 0x1e, 0x7b, 0xdc, 0xec, 0xb4, 0xe2, 0xec, 0xc8, 0x76, 0x93, 0x2f, 
+	0x00, 0x00, 0x00, 0xb0, 0x82, 0x5b, 0x80, 0x01, 0x00, 0x00, 0x00, 0xa0, 
+	0x31, 0x2b, 0x00, 0x01, 0x00, 0x60, 0xc0, 0x8a, 0xd5, 0xfc, 0x77, 0x57, 
+	0xee, 0xcf, 0xac, 0xf0, 0xaf, 0x7a, 0xee, 0x77, 0xe3, 0x77, 0x37, 0x68, 
+	0x24, 0x3a, 0x90, 0xa4, 0xcb, 0x75, 0x00, 0x00, 0x9f, 0x58, 0x01, 0x08, 
+	0x00, 0x00, 0x00, 0x00, 0x8d, 0x29, 0x00, 0x02, 0x00, 0x00, 0x00, 0x40, 
+	0x63, 0x6e, 0x01, 0x06, 0x00, 0x00, 0x80, 0x17, 0xd9, 0x2d, 0xe5, 0xf6, 
+	0x94, 0x7e, 0x46, 0x05, 0x45, 0xee, 0x76, 0x5b, 0xff, 0x29, 0xd7, 0x03, 
+	0x73, 0x14, 0x00, 0xbd, 0xd9, 0xa6, 0xfb, 0xf9, 0xe9, 0xb8, 0x4f, 0x27, 
+	0xbb, 0xa8, 0xc7, 0x67, 0x8f, 0xb3, 0xeb, 0x01, 0xea, 0xde, 0x73, 0xab, 
+	0xaf, 0xeb, 0x9d, 0xce, 0xe1, 0x6d, 0x73, 0x98, 0xf9, 0x0c, 0x00, 0x00, 
+	0xe2, 0x28, 0x00, 0x02, 0x00, 0x40, 0x90, 0xe8, 0x50, 0x8c, 0x99, 0xe7, 
+	0x7e, 0xd7, 0x97, 0xbb, 0xfd, 0x9b, 0xe9, 0xcb, 0x8a, 0x60, 0x90, 0xbb, 
+	0xc7, 0xad, 0x0a, 0x10, 0xa9, 0xba, 0x0e, 0x00, 0xe0, 0x13, 0x7b, 0x00, 
+	0x02, 0x00, 0x00, 0x00, 0x40, 0x63, 0x0a, 0x80, 0x00, 0x00, 0x00, 0x00, 
+	0xd0, 0x98, 0x02, 0x20, 0x00, 0x00, 0x00, 0x00, 0x34, 0x66, 0x0f, 0x40, 
+	0x00, 0x00, 0x00, 0xa0, 0x4c, 0x54, 0x50, 0x64, 0xb6, 0xa8, 0x20, 0xca, 
+	0x53, 0xd2, 0x81, 0xe9, 0x45, 0x01, 0x90, 0xe5, 0x93, 0xf5, 0x29, 0x93, 
+	0xac, 0x49, 0x19, 0x00, 0x38, 0xfd, 0xfb, 0xd8, 0x9d, 0xef, 0x37, 0x33, 
+	0x21, 0x1b, 0x33, 0xdf, 0xe7, 0xa2, 0xc3, 0x38, 0x56, 0x04, 0x88, 0x00, 
+	0xc0, 0xa9, 0x14, 0x00, 0x01, 0xf0, 0x4f, 0xf3, 0x83, 0x7f, 0x26, 0x23, 
+	0xfe, 0x11, 0x5e, 0xd9, 0x26, 0x00, 0x00, 0x80, 0x3d, 0x00, 0x01, 0x00, 
+	0x00, 0x00, 0xa0, 0x31, 0x05, 0x40, 0x00, 0x00, 0x00, 0x00, 0x68, 0x4c, 
+	0x01, 0x10, 0x00, 0x00, 0x00, 0x00, 0x1a, 0xb3, 0x07, 0x20, 0x00, 0x00, 
+	0x1c, 0xee, 0x6e, 0x60, 0xc5, 0xdd, 0xb0, 0x8b, 0x9d, 0x1e, 0x77, 0xf7, 
+	0x7c, 0xa3, 0xfb, 0x02, 0xf4, 0x99, 0x0b, 0x9f, 0xee, 0xb3, 0xbc, 0x5b, 
+	0xda, 0xaf, 0x39, 0x8a, 0x08, 0x0a, 0x80, 0xa4, 0x4d, 0x52, 0x5d, 0xfb, 
+	0x6f, 0xf2, 0x05, 0x00, 0x00, 0x00, 0x4e, 0xe2, 0x16, 0x60, 0x00, 0x00, 
+	0x00, 0x00, 0x68, 0xcc, 0x0a, 0x40, 0x00, 0x08, 0x76, 0xb5, 0x52, 0x78, 
+	0x74, 0x35, 0xb2, 0xd5, 0xc7, 0x00, 0x00, 0xc0, 0x28, 0x2b, 0x00, 0x01, 
+	0x00, 0x00, 0x00, 0xa0, 0x31, 0x2b, 0x00, 0x01, 0x00, 0x20, 0x48, 0xd5, 
+	0x6a, 0xdd, 0xe8, 0x60, 0x8b, 0x99, 0xe7, 0xce, 0xec, 0xbb, 0x3c, 0x13, 
+	0x66, 0x32, 0x33, 0x2e, 0x3b, 0x8d, 0x1f, 0x00, 0x64, 0x50, 0x00, 0x04, 
+	0x00, 0x00, 0x80, 0x17, 0xd9, 0xad, 0x48, 0x5d, 0x15, 0xe4, 0xf8, 0xa9, 
+	0xfd, 0xa7, 0xa9, 0xbe, 0xd9, 0xe3, 0xbf, 0x5b, 0x7f, 0x38, 0x93, 0x02, 
+	0xa0, 0xc9, 0x37, 0x6d, 0x92, 0xad, 0x3a, 0xdf, 0xa8, 0x08, 0xf6, 0xec, 
+	0x49, 0xd6, 0xe4, 0x0b, 0x00, 0x00, 0x00, 0xac, 0x60, 0x0f, 0x40, 0x00, 
+	0x00, 0x00, 0x00, 0x68, 0x4c, 0x01, 0x10, 0x00, 0x00, 0x00, 0x00, 0x1a, 
+	0x73, 0x0b, 0x30, 0x00, 0x29, 0x4e, 0xb9, 0x9d, 0xfd, 0xaa, 0x9f, 0x19, 
+	0x5b, 0x22, 0x8c, 0x1e, 0xef, 0xea, 0x6f, 0x57, 0x6d, 0xae, 0x7e, 0x1d, 
+	0x6c, 0x63, 0xc0, 0x9b, 0x64, 0x6f, 0x9b, 0xb2, 0xdb, 0xb9, 0xdd, 0x7d, 
+	0x7f, 0xdf, 0x0d, 0xd9, 0xd8, 0xe9, 0xb8, 0xae, 0x03, 0x00, 0xba, 0xb3, 
+	0x02, 0x10, 0x00, 0x00, 0x00, 0x00, 0x1a, 0xb3, 0x02, 0x10, 0x00, 0x00, 
+	0x00, 0x1a, 0x3a, 0x65, 0x85, 0x6a, 0x54, 0x90, 0x63, 0x55, 0xfb, 0xd9, 
+	0xe7, 0xfb, 0xb6, 0xeb, 0x81, 0x1c, 0x0a, 0x80, 0x26, 0xdf, 0xdb, 0x76, 
+	0x9b, 0x7c, 0xab, 0xd2, 0x78, 0xa3, 0xfa, 0x63, 0xf2, 0x05, 0x00, 0x00, 
+	0x00, 0x56, 0x70, 0x0b, 0x30, 0x00, 0x00, 0x00, 0x00, 0x34, 0x66, 0x05, 
+	0x20, 0x00, 0x00, 0xfc, 0x60, 0xf7, 0x80, 0x9b, 0x99, 0xa0, 0x8c, 0xdd, 
+	0xc7, 0xef, 0x6e, 0x30, 0xc8, 0xcc, 0xdd, 0x15, 0xbb, 0x8f, 0x1f, 0x00, 
+	0xcc, 0xb2, 0x02, 0x10, 0x00, 0x00, 0x00, 0x00, 0x1a, 0xb3, 0x02, 0x10, 
+	0x80, 0x61, 0x1d, 0xf6, 0xb2, 0xbc, 0x3a, 0x87, 0xab, 0x95, 0x1e, 0xa3, 
+	0xe7, 0x7e, 0xca, 0xf1, 0xba, 0xbf, 0xee, 0x00, 0x00, 0xf0, 0x26, 0x0a, 
+	0x80, 0x00, 0x00, 0x00, 0x70, 0xb0, 0xd3, 0x6f, 0x4f, 0xaf, 0x0a, 0x84, 
+	0x8c, 0x6a, 0x3f, 0xfb, 0xf1, 0x6f, 0xbb, 0x1e, 0xc8, 0xa1, 0x00, 0xe8, 
+	0xcd, 0xb6, 0xcd, 0x64, 0x5a, 0xf5, 0x78, 0xd7, 0x03, 0x00, 0x00, 0x00, 
+	0xd0, 0x99, 0x02, 0x20, 0x00, 0x00, 0xfc, 0x60, 0xc5, 0x0f, 0x89, 0xd1, 
+	0xc1, 0x16, 0x33, 0xc7, 0x9d, 0x19, 0x83, 0xe8, 0x3e, 0xdf, 0x1d, 0x97, 
+	0xe8, 0x1f, 0x5d, 0x67, 0xc2, 0x47, 0x56, 0x3c, 0x0e, 0x00, 0x9e, 0x10, 
+	0x02, 0x02, 0x00, 0x00, 0x00, 0x00, 0x8d, 0x29, 0x00, 0x02, 0x00, 0x00, 
+	0x00, 0x40, 0x63, 0x0a, 0x80, 0x00, 0x00, 0x00, 0x00, 0xd0, 0x98, 0x3d, 
+	0x00, 0x01, 0xb8, 0xd4, 0x7d, 0xdf, 0xa1, 0xab, 0xf3, 0xbb, 0xda, 0xc7, 
+	0x6a, 0xa7, 0x71, 0xc9, 0xd8, 0x23, 0xcc, 0x7e, 0x53, 0x00, 0x70, 0x8e, 
+	0xa7, 0xdf, 0x05, 0xb2, 0x83, 0x1f, 0xb3, 0xbf, 0x77, 0x64, 0xa7, 0xf7, 
+	0x46, 0x8d, 0x67, 0x76, 0xd0, 0xe5, 0xe9, 0xd7, 0x0f, 0x6b, 0x29, 0x00, 
+	0x1e, 0xf8, 0xe6, 0xa9, 0x9a, 0x64, 0xa3, 0xda, 0xdf, 0xad, 0xff, 0x55, 
+	0xa9, 0xc1, 0xbb, 0xa5, 0x18, 0x9b, 0xc4, 0x01, 0x20, 0xe7, 0x73, 0x72, 
+	0x26, 0x3c, 0xa3, 0x2a, 0x4c, 0x62, 0x26, 0x18, 0x64, 0x26, 0xcc, 0x64, 
+	0x45, 0x10, 0xca, 0xdd, 0x3e, 0xaf, 0x18, 0xd3, 0xe8, 0x6b, 0x03, 0x00, 
+	0x3e, 0x71, 0x0b, 0x30, 0x00, 0x00, 0x00, 0x00, 0x34, 0xa6, 0x00, 0x08, 
+	0x00, 0x00, 0x00, 0x00, 0x8d, 0x29, 0x00, 0x02, 0x00, 0x00, 0x00, 0x40, 
+	0x63, 0x0a, 0x80, 0x00, 0x00, 0x00, 0x00, 0xd0, 0x98, 0x10, 0x10, 0x00, 
+	0x00, 0x18, 0xb0, 0x22, 0xd8, 0x22, 0x3a, 0x64, 0x63, 0xe6, 0x3c, 0xa2, 
+	0xcf, 0x2d, 0x3a, 0x90, 0x64, 0x66, 0xfc, 0xa2, 0xc7, 0x25, 0xfa, 0xda, 
+	0x10, 0xf8, 0xc1, 0xe8, 0xfb, 0xe4, 0xd3, 0xb5, 0x18, 0x15, 0x5c, 0x99, 
+	0x9d, 0x26, 0x9c, 0x9d, 0xd2, 0x5b, 0xf5, 0xba, 0x64, 0x7f, 0x0e, 0x55, 
+	0xa5, 0x1b, 0xef, 0x36, 0xfe, 0xfc, 0x3f, 0x05, 0xc0, 0x8d, 0x27, 0xe5, 
+	0xa8, 0x76, 0xb2, 0x8f, 0x9b, 0x1d, 0xb5, 0x9e, 0x9d, 0x7e, 0x1b, 0x35, 
+	0x49, 0x45, 0xf5, 0xb3, 0xea, 0x7c, 0x21, 0xfa, 0x9a, 0xcc, 0xf8, 0xa0, 
+	0xcf, 0x78, 0x7f, 0x8c, 0xf6, 0xf3, 0xaa, 0x2f, 0x19, 0x6d, 0x9e, 0xf2, 
+	0xa5, 0xaa, 0xc3, 0x39, 0x00, 0x00, 0x40, 0x37, 0x6e, 0x01, 0x06, 0x00, 
+	0x00, 0x00, 0x80, 0xc6, 0x14, 0x00, 0x01, 0x00, 0x00, 0x00, 0xa0, 0x31, 
+	0x05, 0x40, 0x00, 0x00, 0x00, 0x00, 0x68, 0xcc, 0x1e, 0x80, 0x00, 0x00, 
+	0x30, 0x60, 0x26, 0x4c, 0x62, 0x45, 0xb0, 0xc5, 0x77, 0x8f, 0x8b, 0x0e, 
+	0xe8, 0xa8, 0x52, 0x15, 0x7a, 0x12, 0x7d, 0xbd, 0x08, 0xfc, 0x00, 0x60, 
+	0x15, 0x05, 0x40, 0x00, 0x00, 0x00, 0x20, 0x2c, 0x40, 0x32, 0xfb, 0xb8, 
+	0x55, 0x41, 0x97, 0x4f, 0xdb, 0x7f, 0xda, 0xff, 0xaa, 0xf1, 0x8f, 0x7a, 
+	0xbd, 0xd8, 0x9b, 0x02, 0xe0, 0x81, 0x93, 0x57, 0xd4, 0xa4, 0x93, 0x3d, 
+	0xe9, 0x47, 0xa5, 0x03, 0x47, 0x4d, 0xa6, 0xbb, 0x7d, 0x68, 0x55, 0x4d, 
+	0xa6, 0x52, 0x86, 0x01, 0x00, 0x00, 0xe0, 0x5d, 0x14, 0x00, 0x01, 0x48, 
+	0x91, 0x51, 0x6c, 0x1e, 0x2d, 0x90, 0x5f, 0xf5, 0xe5, 0xea, 0x6f, 0x57, 
+	0xc7, 0x1b, 0x3d, 0xbf, 0xd5, 0xe7, 0xb0, 0xba, 0x2f, 0xab, 0xaf, 0x17, 
+	0xbf, 0x40, 0x03, 0x00, 0xc0, 0xcf, 0x84, 0x80, 0x00, 0x00, 0x00, 0x00, 
+	0x40, 0x63, 0x56, 0x00, 0x02, 0x00, 0xc0, 0x0f, 0x66, 0xc2, 0x33, 0x66, 
+	0xc2, 0x1f, 0xa2, 0x57, 0xdc, 0xce, 0x1c, 0x23, 0xba, 0x2f, 0x33, 0x63, 
+	0xb0, 0x22, 0x80, 0x25, 0xfa, 0x35, 0x12, 0xf8, 0x01, 0x40, 0x25, 0x2b, 
+	0x00, 0x01, 0x00, 0x00, 0x00, 0xa0, 0x31, 0x2b, 0x00, 0x01, 0x00, 0x00, 
+	0xa0, 0xa1, 0xec, 0xa0, 0xc2, 0xec, 0xa0, 0xcb, 0xaa, 0xa0, 0xc5, 0xec, 
+	0xb4, 0xdf, 0xa8, 0x15, 0xd5, 0xd9, 0xed, 0x3f, 0x3d, 0x2f, 0xf6, 0xa6, 
+	0x00, 0xb8, 0x81, 0xd3, 0x27, 0xbb, 0xaa, 0xf4, 0xde, 0xaa, 0x68, 0xf6, 
+	0xec, 0x49, 0xb0, 0x2a, 0xb5, 0x19, 0x00, 0x00, 0x00, 0xe8, 0xc9, 0x2d, 
+	0xc0, 0x00, 0x00, 0x00, 0x00, 0xd0, 0x98, 0x15, 0x80, 0x00, 0x5c, 0xba, 
+	0x5a, 0xdd, 0x9a, 0xb1, 0xd2, 0x34, 0xe3, 0x78, 0xa3, 0x6d, 0x5e, 0xfd, 
+	0x6d, 0x75, 0x3f, 0x47, 0x8f, 0x97, 0x71, 0x8b, 0xc6, 0x68, 0x9b, 0x19, 
+	0xe3, 0xb9, 0xfa, 0xdc, 0x21, 0xfa, 0xda, 0xff, 0xe9, 0x71, 0xd1, 0xd7, 
+	0xf1, 0x8a, 0xe0, 0x8d, 0xe8, 0x31, 0x98, 0x79, 0xdc, 0xcc, 0xb9, 0xad, 
+	0x18, 0x97, 0xe8, 0x73, 0x03, 0x80, 0x4f, 0xac, 0x00, 0x04, 0x00, 0x00, 
+	0x00, 0x80, 0xc6, 0x14, 0x00, 0x01, 0x00, 0x00, 0x00, 0xa0, 0x31, 0xb7, 
+	0x00, 0x03, 0x00, 0x00, 0xc0, 0xc1, 0x9e, 0x06, 0x15, 0x66, 0xdf, 0x42, 
+	0x9e, 0x9d, 0x42, 0xfb, 0x49, 0x55, 0x00, 0xe6, 0x6e, 0xfd, 0x7c, 0xaa, 
+	0xea, 0x7a, 0xb0, 0x95, 0xc1, 0x5a, 0x0a, 0x80, 0x1b, 0x4c, 0x76, 0xd9, 
+	0x69, 0xb3, 0x51, 0xed, 0x54, 0xa5, 0xfd, 0x66, 0xab, 0x4a, 0x13, 0xae, 
+	0x62, 0x92, 0x05, 0x00, 0x00, 0x80, 0x77, 0x51, 0x00, 0x04, 0x00, 0x80, 
+	0x1f, 0xac, 0xf8, 0x01, 0x6d, 0x26, 0x74, 0x22, 0x3a, 0x9c, 0x62, 0x45, 
+	0xe0, 0xc7, 0xcc, 0x71, 0xab, 0x8e, 0xb1, 0xe2, 0xb9, 0x55, 0xd7, 0x1f, 
+	0x00, 0xbd, 0xd9, 0x03, 0x10, 0x00, 0x00, 0x00, 0x00, 0x1a, 0xb3, 0x02, 
+	0x10, 0x80, 0x14, 0x57, 0xab, 0x15, 0x32, 0x6e, 0xf9, 0xcf, 0x58, 0x1d, 
+	0x91, 0x71, 0x0e, 0xa3, 0xfd, 0xbc, 0x3a, 0xde, 0xea, 0xb1, 0xee, 0xf0, 
+	0x3a, 0x00, 0x00, 0xc0, 0x9b, 0x58, 0x01, 0x08, 0x00, 0x00, 0x00, 0x00, 
+	0x8d, 0x59, 0x01, 0x08, 0x00, 0x00, 0x00, 0x07, 0xab, 0x0a, 0x96, 0x8c, 
+	0xea, 0x4f, 0x76, 0x3b, 0x55, 0xa9, 0xc4, 0xa7, 0xdc, 0xa9, 0x50, 0x75, 
+	0x9d, 0xb0, 0x96, 0x02, 0xe0, 0xc6, 0x93, 0xf2, 0x6e, 0x1f, 0x06, 0xa7, 
+	0xa4, 0x1b, 0x57, 0xf5, 0xe7, 0x94, 0x49, 0x50, 0x04, 0x3b, 0x00, 0xac, 
+	0xfd, 0x9c, 0xbd, 0x1b, 0xda, 0x31, 0x13, 0x1c, 0x11, 0x1d, 0x4e, 0x31, 
+	0xf3, 0xb8, 0x15, 0xdf, 0x33, 0x66, 0xc6, 0x74, 0xc5, 0xf8, 0x55, 0x85, 
+	0x85, 0x00, 0xc0, 0x27, 0x6e, 0x01, 0x06, 0x00, 0x00, 0x00, 0x80, 0xc6, 
+	0x14, 0x00, 0x01, 0x00, 0x00, 0x00, 0xa0, 0x31, 0x05, 0x40, 0x00, 0x00, 
+	0x00, 0x00, 0x68, 0xcc, 0x1e, 0x80, 0x00, 0x5c, 0xca, 0xd8, 0xc7, 0x72, 
+	0x74, 0x7f, 0xa3, 0xab, 0xe7, 0x5d, 0xf5, 0xf3, 0xea, 0x79, 0xa3, 0x6d, 
+	0x66, 0xf4, 0xb3, 0x43, 0x9b, 0x19, 0xaf, 0x3b, 0x00, 0x00, 0x30, 0x47, 
+	0x01, 0x10, 0x00, 0x00, 0x7e, 0x30, 0x13, 0xd6, 0x10, 0xfd, 0xdc, 0xbb, 
+	0xa2, 0x8b, 0xee, 0xd1, 0x3f, 0x0c, 0x44, 0x07, 0x92, 0xcc, 0x84, 0x94, 
+	0xcc, 0x84, 0x85, 0xcc, 0x3c, 0x77, 0xe6, 0x7c, 0x21, 0xe3, 0xfd, 0x59, 
+	0xd5, 0x4e, 0x76, 0x60, 0xe6, 0x6e, 0x81, 0x9c, 0x51, 0xe3, 0xb9, 0x5b, 
+	0xa0, 0xa5, 0xb9, 0x6a, 0x6f, 0x0a, 0x80, 0x0b, 0xdf, 0xe4, 0x55, 0xa9, 
+	0xb5, 0xd9, 0x93, 0x66, 0xd4, 0x64, 0x51, 0x35, 0xb9, 0xef, 0x16, 0x6d, 
+	0x5e, 0x75, 0xfd, 0x00, 0x00, 0x00, 0x00, 0x3d, 0xd9, 0x03, 0x10, 0x00, 
+	0x00, 0x00, 0x00, 0x1a, 0x53, 0x00, 0x04, 0x00, 0x00, 0x00, 0x80, 0xc6, 
+	0x14, 0x00, 0x01, 0x00, 0x00, 0x00, 0xa0, 0x31, 0x7b, 0x00, 0x02, 0x00, 
+	0xc0, 0x80, 0xe8, 0xb0, 0x86, 0x15, 0xe1, 0x0f, 0x77, 0xdb, 0x8b, 0x0e, 
+	0xa7, 0x58, 0x31, 0x56, 0xd1, 0x76, 0x1a, 0x03, 0xc1, 0x20, 0x00, 0xcc, 
+	0x52, 0x00, 0x04, 0x20, 0xe5, 0x1f, 0xc9, 0xab, 0x7f, 0xce, 0x46, 0xff, 
+	0x69, 0x19, 0x6d, 0x33, 0xa3, 0x2f, 0x3b, 0x8d, 0x67, 0xc6, 0x3f, 0xd0, 
+	0xab, 0xfb, 0x92, 0xd1, 0x4f, 0x00, 0xf0, 0x5d, 0xad, 0x26, 0x58, 0xf2, 
+	0x69, 0x3b, 0xd9, 0x69, 0xb3, 0xa7, 0xa7, 0xfd, 0x46, 0xb5, 0x53, 0x75, 
+	0x3d, 0x64, 0xa7, 0x15, 0x73, 0x8f, 0x02, 0xe0, 0x81, 0x93, 0x72, 0x55, 
+	0xfb, 0xbb, 0xa5, 0xf4, 0xee, 0x16, 0x75, 0x9f, 0xfd, 0x61, 0xb0, 0xdb, 
+	0x87, 0x1c, 0x00, 0x00, 0x00, 0x70, 0x06, 0x7b, 0x00, 0x02, 0x00, 0x00, 
+	0x00, 0x40, 0x63, 0x0a, 0x80, 0x00, 0x00, 0x00, 0x00, 0xd0, 0x98, 0x5b, 
+	0x80, 0x01, 0x00, 0xe0, 0x07, 0x2b, 0xc2, 0x1a, 0x66, 0x42, 0x27, 0xee, 
+	0x6e, 0xfd, 0x31, 0xf3, 0xdc, 0xbb, 0xe7, 0x1b, 0x7d, 0xdc, 0xe8, 0xe7, 
+	0x46, 0x9f, 0x5b, 0x55, 0x20, 0x09, 0x00, 0x3c, 0x61, 0x05, 0x20, 0x00, 
+	0x00, 0x00, 0x00, 0x34, 0x66, 0x05, 0x20, 0x00, 0x00, 0x00, 0x1c, 0x2c, 
+	0x3b, 0x38, 0x31, 0xea, 0xb8, 0x55, 0xfd, 0xcc, 0xee, 0xff, 0x27, 0x4f, 
+	0xdb, 0xa9, 0x5a, 0xed, 0x5b, 0x15, 0x4c, 0xca, 0x5a, 0x0a, 0x80, 0x1b, 
+	0x38, 0x3d, 0xcd, 0xb6, 0x6a, 0x72, 0xcf, 0x8e, 0xa8, 0x8f, 0x9a, 0xc4, 
+	0xb3, 0xdb, 0x31, 0xf9, 0x72, 0xda, 0x17, 0xd1, 0x99, 0xb9, 0xec, 0xaa, 
+	0xcd, 0xd1, 0xe7, 0x9d, 0xf2, 0x79, 0xb0, 0xfa, 0xdc, 0x47, 0xe7, 0xea, 
+	0xd1, 0xe7, 0xad, 0xbe, 0x5e, 0x00, 0x00, 0xe0, 0x4d, 0xdc, 0x02, 0x0c, 
+	0x00, 0x00, 0x00, 0x00, 0x8d, 0x59, 0x01, 0x08, 0x00, 0x00, 0x3f, 0xb8, 
+	0x1b, 0xc6, 0x11, 0x1d, 0xe4, 0x31, 0xf3, 0xb8, 0xbb, 0xe7, 0xf1, 0x9d, 
+	0x15, 0x2b, 0x68, 0x77, 0x0a, 0x33, 0x89, 0x1e, 0x83, 0x15, 0xaf, 0x25, 
+	0x00, 0x3c, 0x61, 0x05, 0x20, 0x00, 0x00, 0x00, 0x00, 0x34, 0xa6, 0x00, 
+	0x08, 0x00, 0x00, 0x00, 0x00, 0x8d, 0xb9, 0x05, 0x18, 0x00, 0x00, 0x00, 
+	0x0e, 0xb6, 0x5b, 0xea, 0xee, 0xd3, 0x5b, 0xd9, 0x9f, 0xf6, 0x33, 0x3b, 
+	0x5d, 0x37, 0x7b, 0xdc, 0xaa, 0x52, 0x8f, 0xa3, 0x5e, 0xc7, 0xa7, 0xaf, 
+	0x4b, 0xd4, 0x75, 0xc2, 0x1c, 0x05, 0xc0, 0x0d, 0x26, 0xdf, 0x53, 0x26, 
+	0xfd, 0xa8, 0xf3, 0xca, 0x9e, 0x44, 0x4e, 0xb9, 0x1e, 0xa2, 0x1e, 0x7f, 
+	0x4a, 0xfa, 0x30, 0x00, 0x00, 0x00, 0x50, 0x43, 0x01, 0x10, 0x80, 0x14, 
+	0x57, 0xc5, 0xec, 0xd1, 0x82, 0x73, 0x46, 0x9b, 0xa3, 0x32, 0x8e, 0x77, 
+	0xd5, 0xe6, 0xe8, 0x8f, 0x21, 0xa3, 0xcf, 0x1b, 0xed, 0xcb, 0xe8, 0xf3, 
+	0x32, 0xda, 0x84, 0x5d, 0xe6, 0xbe, 0x13, 0xc3, 0x24, 0x56, 0x04, 0x6a, 
+	0x44, 0xf7, 0xb9, 0x2a, 0x18, 0x64, 0xe6, 0x18, 0x82, 0x41, 0x00, 0x58, 
+	0xc5, 0x1e, 0x80, 0x00, 0x00, 0x00, 0x00, 0xd0, 0x98, 0x02, 0x20, 0x00, 
+	0x00, 0x00, 0x00, 0x34, 0xa6, 0x00, 0x08, 0x00, 0x00, 0x00, 0x00, 0x8d, 
+	0xd9, 0x03, 0x10, 0x00, 0x00, 0x00, 0x1a, 0xca, 0x0e, 0x72, 0x7c, 0x7a, 
+	0xdc, 0xa8, 0xf6, 0xb3, 0xfb, 0xd3, 0x55, 0x76, 0xda, 0xb2, 0xf1, 0xdf, 
+	0x9b, 0x02, 0xe0, 0xc2, 0x37, 0x55, 0xd4, 0x64, 0x5d, 0xd5, 0xce, 0xd3, 
+	0xc9, 0xa2, 0x6a, 0x12, 0xaf, 0x8a, 0xba, 0xcf, 0xbe, 0x1e, 0x4e, 0x1f, 
+	0x07, 0x00, 0xf0, 0x3d, 0x70, 0xbf, 0xef, 0x11, 0x9f, 0x8e, 0x11, 0x1d, 
+	0x3e, 0x32, 0xf3, 0xb8, 0x9d, 0x5e, 0xb7, 0xbb, 0xfd, 0x8b, 0x0e, 0xf7, 
+	0x10, 0x0c, 0x02, 0xc0, 0x2c, 0xb7, 0x00, 0x03, 0x00, 0x00, 0x00, 0x40, 
+	0x63, 0x56, 0x00, 0x02, 0x90, 0xe2, 0x6a, 0x65, 0xc2, 0xd5, 0x0a, 0x8a, 
+	0xd1, 0xe7, 0x8d, 0xae, 0x1a, 0x19, 0x5d, 0x41, 0xd1, 0x7d, 0x35, 0xed, 
+	0xe8, 0xf9, 0x65, 0xbc, 0xb6, 0x19, 0xaf, 0x1f, 0x00, 0x00, 0xbc, 0x89, 
+	0x15, 0x80, 0x00, 0x00, 0x00, 0x00, 0xd0, 0x98, 0x02, 0x20, 0x00, 0x00, 
+	0x00, 0x00, 0x34, 0xe6, 0x16, 0x60, 0x00, 0x00, 0xd8, 0x54, 0x55, 0xe8, 
+	0xc4, 0xcc, 0xed, 0xf5, 0x77, 0xfb, 0x1c, 0x1d, 0x6c, 0xb1, 0x7b, 0x7b, 
+	0x33, 0x6c, 0x77, 0x40, 0xf4, 0xb5, 0x53, 0x95, 0xd2, 0x5b, 0x95, 0x36, 
+	0x5b, 0x15, 0x38, 0xb9, 0x5b, 0x7a, 0x72, 0xd5, 0x1c, 0x63, 0x0e, 0xdb, 
+	0x83, 0x02, 0x60, 0xa3, 0x49, 0x3c, 0xea, 0xcb, 0x59, 0xf6, 0xa4, 0x93, 
+	0x9d, 0x5a, 0x5b, 0xb5, 0x2f, 0x57, 0xf6, 0x79, 0x7d, 0x6a, 0x47, 0xda, 
+	0x2f, 0x00, 0x00, 0x00, 0x70, 0xc5, 0x2d, 0xc0, 0x00, 0x00, 0x00, 0x00, 
+	0xd0, 0x98, 0x02, 0x20, 0x00, 0x00, 0x00, 0x00, 0x34, 0xe6, 0x16, 0x60, 
+	0x00, 0x2e, 0x65, 0xec, 0xd9, 0x31, 0xda, 0x66, 0x46, 0x5f, 0xae, 0x6e, 
+	0x97, 0xbf, 0x3a, 0xde, 0xe8, 0xf3, 0x56, 0xf7, 0x65, 0xa7, 0xf1, 0x5c, 
+	0xdd, 0x17, 0x00, 0x00, 0xe0, 0x5f, 0x0a, 0x80, 0x00, 0x00, 0x10, 0x24, 
+	0x3a, 0x38, 0xe2, 0xee, 0x73, 0x67, 0x82, 0x37, 0x66, 0x8e, 0x1b, 0x7d, 
+	0x1e, 0x33, 0xe3, 0x5c, 0xf5, 0x1a, 0xed, 0xd4, 0x17, 0x00, 0xf8, 0x44, 
+	0x01, 0x10, 0x00, 0x00, 0x00, 0x5e, 0x24, 0x3b, 0xa5, 0xb7, 0x2a, 0x58, 
+	0xb2, 0x6a, 0x7c, 0xb2, 0xd3, 0x81, 0x4f, 0x19, 0x87, 0xa8, 0xf6, 0xc9, 
+	0xa1, 0x00, 0x98, 0x60, 0xb7, 0x14, 0xda, 0xec, 0x49, 0xed, 0x69, 0x7f, 
+	0x76, 0xfb, 0x90, 0x38, 0x65, 0x52, 0x7e, 0xda, 0x4f, 0xe9, 0xc0, 0x00, 
+	0x00, 0x00, 0xc0, 0xd7, 0x97, 0x10, 0x10, 0x00, 0x00, 0x00, 0x00, 0x68, 
+	0x4d, 0x01, 0x10, 0x00, 0x00, 0x00, 0x00, 0x1a, 0x73, 0x0b, 0x30, 0x00, 
+	0x00, 0x2c, 0x36, 0x13, 0xda, 0xb1, 0x62, 0xcf, 0xa4, 0x15, 0x5b, 0x86, 
+	0x44, 0x07, 0x92, 0x7c, 0xf7, 0xb8, 0xe8, 0x70, 0x94, 0xe8, 0x71, 0x99, 
+	0x09, 0x79, 0xb1, 0x77, 0x16, 0x00, 0x4f, 0x28, 0x00, 0x02, 0x50, 0xfe, 
+	0x4f, 0x60, 0xb6, 0x8c, 0x7f, 0x92, 0xae, 0xda, 0x1c, 0x1d, 0xb3, 0xd1, 
+	0x36, 0x33, 0xfa, 0x32, 0x7a, 0xbc, 0x9d, 0xae, 0x4f, 0xff, 0x1c, 0x03, 
+	0x00, 0xc0, 0xbf, 0x14, 0x00, 0x01, 0x00, 0x00, 0xe0, 0x60, 0x51, 0x3f, 
+	0x7a, 0x55, 0xb5, 0x13, 0x15, 0x78, 0xf8, 0xb4, 0xfd, 0xec, 0x71, 0x88, 
+	0x1a, 0xb7, 0xaa, 0xe0, 0xca, 0xec, 0xe3, 0xb2, 0x96, 0x02, 0xe0, 0xc6, 
+	0x93, 0xef, 0x6e, 0xc7, 0x3d, 0x25, 0x65, 0x78, 0xb7, 0xf3, 0x8d, 0x3a, 
+	0xae, 0x49, 0x16, 0x00, 0x00, 0x00, 0x18, 0x21, 0x04, 0x04, 0x00, 0x00, 
+	0x00, 0x00, 0x1a, 0xb3, 0x02, 0x10, 0x00, 0x00, 0x7e, 0x30, 0x13, 0x44, 
+	0x31, 0xf3, 0xb8, 0x15, 0x7d, 0x9e, 0x09, 0x98, 0xb8, 0xfb, 0xdc, 0x99, 
+	0x3b, 0x37, 0xa2, 0x03, 0x3a, 0xee, 0xf6, 0x6f, 0xe6, 0x35, 0x9a, 0x19, 
+	0x17, 0x77, 0x7d, 0x00, 0x90, 0xc1, 0x0a, 0x40, 0x00, 0x00, 0x00, 0x00, 
+	0x68, 0x4c, 0x01, 0x10, 0x00, 0x00, 0x00, 0x00, 0x1a, 0x73, 0x0b, 0x30, 
+	0x00, 0xc3, 0x76, 0xba, 0x4d, 0x29, 0x3b, 0x18, 0xa8, 0xfa, 0x78, 0x57, 
+	0x63, 0x9d, 0xd1, 0x97, 0x53, 0x6e, 0x41, 0x73, 0xab, 0x1c, 0x00, 0xac, 
+	0xff, 0x5e, 0xb2, 0xea, 0x73, 0x3b, 0x2a, 0x20, 0xf1, 0x69, 0x40, 0xe5, 
+	0xd3, 0xf1, 0x7c, 0xda, 0xcf, 0xaa, 0x40, 0x4b, 0xc1, 0x95, 0xef, 0xa6, 
+	0x00, 0x68, 0xf2, 0x9d, 0x9e, 0xa4, 0xa2, 0x26, 0xb5, 0xec, 0x71, 0x3b, 
+	0x7d, 0xf2, 0x3d, 0x25, 0xba, 0x1e, 0x00, 0x00, 0x00, 0xd8, 0x8b, 0x02, 
+	0x20, 0x00, 0x00, 0xfc, 0x60, 0xa7, 0x1f, 0x78, 0xab, 0xc2, 0x29, 0xee, 
+	0xf6, 0x65, 0x66, 0xac, 0xaa, 0x82, 0x4b, 0xa2, 0xc7, 0x2f, 0xfa, 0x5a, 
+	0xf3, 0xc3, 0x2e, 0x00, 0xb3, 0xec, 0x01, 0x08, 0x00, 0x00, 0x00, 0x00, 
+	0x8d, 0x29, 0x00, 0x02, 0x00, 0x00, 0x00, 0x40, 0x63, 0x0a, 0x80, 0x00, 
+	0x00, 0x00, 0x00, 0xd0, 0x98, 0x3d, 0x00, 0x01, 0x00, 0x00, 0xa0, 0xa1, 
+	0xaa, 0xfd, 0x23, 0xa3, 0xf6, 0x4d, 0xcd, 0x4e, 0xad, 0x7d, 0x7a, 0xdc, 
+	0xaa, 0xf3, 0xaa, 0x0a, 0x84, 0xac, 0x4a, 0x6d, 0x26, 0x87, 0x02, 0xa0, 
+	0xc9, 0x77, 0xf9, 0xe3, 0xa3, 0x52, 0x86, 0x9f, 0xf6, 0xa7, 0x6a, 0xf3, 
+	0xee, 0xaa, 0xc9, 0x77, 0xb7, 0xeb, 0x90, 0xf7, 0xc9, 0x78, 0xcf, 0x65, 
+	0x5c, 0xbf, 0x57, 0x6d, 0x8e, 0x9e, 0xc3, 0x68, 0x9b, 0xab, 0xe7, 0xa9, 
+	0x9d, 0x42, 0x0d, 0xae, 0xfa, 0x62, 0xde, 0xe2, 0xcd, 0xdf, 0xed, 0x56, 
+	0x84, 0x71, 0xac, 0x38, 0xc6, 0xee, 0xe1, 0x1e, 0x2b, 0x5e, 0xf3, 0xe8, 
+	0x71, 0x36, 0x37, 0x02, 0xf0, 0x84, 0x5b, 0x80, 0x01, 0x00, 0x00, 0x00, 
+	0xa0, 0x31, 0x05, 0x40, 0x00, 0x00, 0x00, 0x00, 0x68, 0x4c, 0x01, 0x10, 
+	0x00, 0x00, 0x00, 0x00, 0x1a, 0x53, 0x00, 0x04, 0x00, 0x00, 0x00, 0x80, 
+	0xc6, 0x84, 0x80, 0x00, 0x00, 0x40, 0x90, 0x99, 0x20, 0x8a, 0xe8, 0x80, 
+	0x89, 0xe8, 0x30, 0x89, 0x15, 0x01, 0x1d, 0x77, 0xdb, 0x8b, 0x0e, 0xc0, 
+	0xf8, 0xae, 0x7f, 0xd1, 0xa1, 0x22, 0x2b, 0xc6, 0x19, 0x56, 0xcc, 0x55, 
+	0x19, 0xd7, 0x66, 0x76, 0x2a, 0x6e, 0xf6, 0xfb, 0x38, 0xaa, 0x3f, 0xd9, 
+	0x81, 0x9c, 0x51, 0xfd, 0xf9, 0x34, 0x0e, 0xd9, 0x81, 0x9f, 0xcc, 0x51, 
+	0x00, 0x34, 0xf9, 0xb6, 0x9b, 0xac, 0xa3, 0x26, 0x9d, 0xd3, 0x27, 0x5f, 
+	0x00, 0x00, 0x00, 0x80, 0xaf, 0x2f, 0x05, 0x40, 0x00, 0x58, 0xea, 0xaa, 
+	0x68, 0x9f, 0xf1, 0x43, 0x42, 0xc6, 0xf1, 0x32, 0x7e, 0x78, 0x38, 0xa5, 
+	0x9f, 0x00, 0x00, 0x70, 0x22, 0x7b, 0x00, 0x02, 0x00, 0x00, 0x00, 0x40, 
+	0x63, 0x0a, 0x80, 0x00, 0x00, 0x00, 0x00, 0xd0, 0x98, 0x5b, 0x80, 0x01, 
+	0x00, 0xe0, 0xc5, 0xaa, 0x42, 0x27, 0xaa, 0x02, 0x44, 0x56, 0x9c, 0xef, 
+	0xdd, 0x50, 0x91, 0xef, 0x8e, 0x6b, 0xfb, 0x02, 0x00, 0x32, 0x28, 0x00, 
+	0x02, 0x00, 0x00, 0x00, 0xdb, 0xd9, 0x2d, 0x08, 0x31, 0xaa, 0x3f, 0x4f, 
+	0xf7, 0x37, 0xae, 0x4a, 0xd7, 0xfd, 0xd4, 0xbe, 0x80, 0xca, 0x33, 0x29, 
+	0x00, 0x32, 0x3d, 0xb9, 0x64, 0xb7, 0xbf, 0x5b, 0xc4, 0xfb, 0x29, 0x1f, 
+	0x4e, 0xa7, 0xf7, 0x13, 0x00, 0x00, 0x00, 0x88, 0x61, 0x0f, 0x40, 0x00, 
+	0x00, 0x00, 0x00, 0x68, 0xcc, 0x0a, 0x40, 0x00, 0x58, 0xe8, 0x6a, 0x55, 
+	0x73, 0xc6, 0x4a, 0xdc, 0xd5, 0xc7, 0x1b, 0xed, 0xcb, 0x95, 0xab, 0x7e, 
+	0x56, 0xad, 0x12, 0x07, 0x00, 0x80, 0x93, 0x28, 0x00, 0x02, 0x00, 0xc0, 
+	0x4b, 0x44, 0x07, 0x6f, 0xac, 0x28, 0xc2, 0xcf, 0xfc, 0x58, 0x71, 0x37, 
+	0x78, 0x23, 0xfa, 0x71, 0x55, 0x01, 0x27, 0x00, 0xf0, 0x89, 0x5b, 0x80, 
+	0x01, 0x00, 0x00, 0x00, 0xa0, 0x31, 0x2b, 0x00, 0x01, 0x00, 0x00, 0x80, 
+	0xed, 0x54, 0xa5, 0xeb, 0xee, 0x76, 0xbe, 0x4f, 0xcf, 0xab, 0x6a, 0x8b, 
+	0x14, 0x81, 0x93, 0x7b, 0x53, 0x00, 0x24, 0x6d, 0x32, 0x7a, 0xfa, 0x26, 
+	0xcf, 0x4e, 0xdd, 0x3d, 0x65, 0xd2, 0x3c, 0xa5, 0x3f, 0x00, 0x00, 0x00, 
+	0xc0, 0x19, 0xdc, 0x02, 0x0c, 0x00, 0x00, 0x00, 0x00, 0x8d, 0x59, 0x01, 
+	0x08, 0x00, 0x00, 0x41, 0x76, 0xba, 0xcd, 0xe9, 0x6e, 0x60, 0xc5, 0x8a, 
+	0x73, 0x8b, 0xbe, 0x93, 0x21, 0xba, 0xbd, 0xbb, 0x81, 0x1f, 0x2b, 0xfa, 
+	0xec, 0x56, 0x39, 0x00, 0x32, 0x28, 0x00, 0x02, 0xd0, 0xfe, 0x9f, 0xe0, 
+	0xd1, 0x7f, 0xc8, 0xae, 0xce, 0x61, 0xf4, 0x1f, 0xc3, 0x0e, 0xe3, 0xd2, 
+	0xfd, 0x9a, 0x00, 0x00, 0x80, 0x6e, 0xdc, 0x02, 0x0c, 0x00, 0x00, 0x00, 
+	0x00, 0x8d, 0x59, 0x01, 0x08, 0x00, 0x00, 0x00, 0x2f, 0xb2, 0xdb, 0xaa, 
+	0xfc, 0xec, 0x80, 0xc7, 0xa8, 0xf3, 0x8d, 0x0a, 0xc6, 0xcc, 0x0e, 0xd8, 
+	0xcc, 0x1e, 0xcf, 0xec, 0xf1, 0x21, 0x87, 0x02, 0xa0, 0xc9, 0x37, 0xed, 
+	0xcd, 0x5f, 0x35, 0x19, 0x45, 0x3d, 0x3e, 0xbb, 0x3f, 0xd9, 0x1f, 0x66, 
+	0x52, 0x83, 0x01, 0x00, 0x00, 0x80, 0xaf, 0x2f, 0x05, 0x40, 0x00, 0x00, 
+	0x18, 0xd2, 0xf9, 0xc7, 0xb6, 0x99, 0x50, 0x8c, 0xef, 0x9e, 0x7b, 0xb7, 
+	0xbd, 0xbb, 0x3f, 0x8c, 0xdf, 0x7d, 0xdc, 0x77, 0xc7, 0x98, 0x39, 0x8f, 
+	0xbb, 0xed, 0xdd, 0x1d, 0x83, 0xbb, 0x7d, 0xb6, 0x4a, 0x06, 0x80, 0x59, 
+	0xf6, 0x00, 0x04, 0x00, 0x00, 0x00, 0x80, 0xc6, 0x14, 0x00, 0x01, 0x00, 
+	0x00, 0x00, 0xa0, 0x31, 0xb7, 0x00, 0x03, 0x30, 0xec, 0xcd, 0x7b, 0x4d, 
+	0x5e, 0xdd, 0x8e, 0x75, 0x35, 0x2e, 0x19, 0xcf, 0x5b, 0x7d, 0x7e, 0x19, 
+	0xd7, 0x8b, 0xdb, 0xdb, 0x00, 0x00, 0x20, 0x8f, 0x02, 0x20, 0x00, 0x00, 
+	0x00, 0x34, 0x74, 0xfa, 0x8f, 0xb5, 0x55, 0x69, 0xb6, 0x55, 0xfd, 0x8f, 
+	0x52, 0x95, 0x32, 0xfc, 0xb4, 0x3f, 0xac, 0xa5, 0x00, 0x68, 0xf2, 0x4d, 
+	0x9b, 0x44, 0x76, 0x8b, 0x72, 0xdf, 0x2d, 0x7a, 0xbc, 0xea, 0x7a, 0x30, 
+	0xf9, 0x02, 0x40, 0xbf, 0xcf, 0xcf, 0x15, 0xfd, 0x8b, 0x3e, 0x86, 0x3e, 
+	0xfb, 0xae, 0x06, 0xc0, 0x3a, 0xf6, 0x00, 0x04, 0x00, 0x00, 0x00, 0x80, 
+	0xc6, 0x14, 0x00, 0x01, 0x00, 0x00, 0x00, 0xa0, 0x31, 0x05, 0x40, 0x00, 
+	0x00, 0x00, 0x00, 0x68, 0x4c, 0x01, 0x10, 0x00, 0x00, 0x00, 0x00, 0x1a, 
+	0x13, 0x02, 0x02, 0xc0, 0xa5, 0xee, 0x9b, 0x91, 0x67, 0x9c, 0xdf, 0x68, 
+	0x9b, 0x3b, 0xf5, 0xa5, 0xfb, 0x58, 0x03, 0x80, 0xef, 0x33, 0xfd, 0xfa, 
+	0x5f, 0x35, 0x0e, 0xa7, 0xf4, 0xff, 0xf4, 0x71, 0x66, 0x8e, 0x02, 0xa0, 
+	0xc9, 0x77, 0xba, 0xff, 0x26, 0x3b, 0xd7, 0x21, 0x00, 0x00, 0x00, 0xb0, 
+	0x2f, 0xb7, 0x00, 0x03, 0x00, 0x00, 0x00, 0x40, 0x63, 0x0a, 0x80, 0x00, 
+	0x00, 0x00, 0x00, 0xd0, 0x98, 0x02, 0x20, 0x00, 0x00, 0x00, 0x00, 0x34, 
+	0xa6, 0x00, 0x08, 0x00, 0x00, 0x00, 0x00, 0x8d, 0x29, 0x00, 0x02, 0x00, 
+	0x00, 0x00, 0x40, 0x63, 0x0a, 0x80, 0x00, 0x00, 0x00, 0x00, 0xd0, 0x98, 
+	0x02, 0x20, 0x00, 0x00, 0x00, 0x00, 0x34, 0xa6, 0x00, 0x08, 0x00, 0x00, 
+	0x00, 0x00, 0x8d, 0x29, 0x00, 0x02, 0x00, 0x00, 0x00, 0x40, 0x63, 0x0a, 
+	0x80, 0x00, 0x00, 0x00, 0x00, 0xd0, 0x98, 0x02, 0x20, 0x00, 0x00, 0x00, 
+	0x00, 0x34, 0xf6, 0x87, 0x21, 0x00, 0x78, 0x97, 0xbf, 0xfe, 0xfe, 0xc7, 
+	0x20, 0x00, 0xe6, 0x3d, 0x00, 0x73, 0x16, 0xf0, 0x22, 0x0a, 0x80, 0x26, 
+	0x5f, 0x00, 0x00, 0x00, 0x00, 0x1a, 0x73, 0x0b, 0x30, 0x00, 0x00, 0x00, 
+	0x00, 0x34, 0xa6, 0x00, 0x08, 0x00, 0x00, 0x00, 0x00, 0x8d, 0x29, 0x00, 
+	0x02, 0x00, 0x00, 0x00, 0x40, 0x63, 0x0a, 0x80, 0x00, 0x00, 0x00, 0x00, 
+	0xd0, 0x98, 0x02, 0x20, 0x00, 0x00, 0x00, 0x00, 0x34, 0xa6, 0x00, 0x08, 
+	0x00, 0x00, 0x00, 0x00, 0x8d, 0x29, 0x00, 0x02, 0x00, 0x00, 0x00, 0x40, 
+	0x63, 0x0a, 0x80, 0x00, 0x00, 0x00, 0x00, 0xd0, 0x98, 0x02, 0x20, 0x00, 
+	0x00, 0x00, 0x00, 0x34, 0xa6, 0x00, 0x08, 0x00, 0x00, 0x00, 0x00, 0x8d, 
+	0x29, 0x00, 0x02, 0x00, 0x00, 0x00, 0x40, 0x63, 0x0a, 0x80, 0x00, 0x00, 
+	0x00, 0x00, 0xd0, 0x98, 0x02, 0x20, 0x00, 0x00, 0x00, 0x00, 0x34, 0xa6, 
+	0x00, 0x08, 0x00, 0x00, 0x00, 0x00, 0x8d, 0x29, 0x00, 0x02, 0x00, 0x00, 
+	0x00, 0x40, 0x63, 0x0a, 0x80, 0x00, 0x00, 0x00, 0x00, 0xd0, 0x98, 0x02, 
+	0x20, 0x00, 0x00, 0x00, 0x00, 0x34, 0xa6, 0x00, 0x08, 0x00, 0x00, 0x00, 
+	0x00, 0x8d, 0x29, 0x00, 0x02, 0x00, 0x00, 0x00, 0x40, 0x63, 0x0a, 0x80, 
+	0x00, 0x00, 0x00, 0x00, 0xd0, 0x98, 0x02, 0x20, 0x00, 0x00, 0x00, 0x00, 
+	0x34, 0xa6, 0x00, 0x08, 0x00, 0x00, 0x00, 0x00, 0x8d, 0x29, 0x00, 0x02, 
+	0x00, 0x00, 0x00, 0x40, 0x63, 0x0a, 0x80, 0x00, 0x00, 0x00, 0x00, 0xd0, 
+	0x98, 0x02, 0x20, 0x00, 0x00, 0x00, 0x00, 0x34, 0xa6, 0x00, 0x08, 0x00, 
+	0x00, 0x00, 0x00, 0x8d, 0x29, 0x00, 0x02, 0x00, 0x00, 0x00, 0x40, 0x63, 
+	0x0a, 0x80, 0x00, 0x00, 0x00, 0x00, 0xd0, 0x98, 0x02, 0x20, 0x00, 0x00, 
+	0x00, 0x00, 0x34, 0xa6, 0x00, 0x08, 0x00, 0x00, 0x00, 0x00, 0x8d, 0x29, 
+	0x00, 0x02, 0x00, 0x00, 0x00, 0x40, 0x63, 0x0a, 0x80, 0x00, 0x00, 0x00, 
+	0x00, 0xd0, 0x98, 0x02, 0x20, 0x00, 0x00, 0x00, 0x00, 0x34, 0xa6, 0x00, 
+	0x08, 0x00, 0x00, 0x00, 0x00, 0x8d, 0x29, 0x00, 0x02, 0x00, 0x00, 0x00, 
+	0x40, 0x63, 0x0a, 0x80, 0x00, 0x00, 0x00, 0x00, 0xd0, 0x98, 0x02, 0x20, 
+	0x00, 0x00, 0x00, 0x00, 0x34, 0xa6, 0x00, 0x08, 0x00, 0x00, 0x00, 0x00, 
+	0x8d, 0xfd, 0x0f, 0xc7, 0x94, 0xeb, 0x3d, 0x40, 0xed, 0x18, 0x19, 0x00, 
+	0x00, 0x00, 0x00, 0x49, 0x45, 0x4e, 0x44, 0xae, 0x42, 0x60, 0x82, 
 	0x00
 };
 
-uint32_t about_png_size = 70711;
+uint32_t about_png_size = 70703;
 
diff --git a/yabause/src/retro_arena/MenuScreen.cpp b/yabause/src/retro_arena/MenuScreen.cpp
index 706f26e3b..e22e8f8c5 100644
--- a/yabause/src/retro_arena/MenuScreen.cpp
+++ b/yabause/src/retro_arena/MenuScreen.cpp
@@ -67,6 +67,24 @@ MenuScreen::MenuScreen( SDL_Window* pwindow, int rwidth, int rheight, const std:
   current_game_path_ = game;
   swindow = nullptr;
   imageWindow = nullptr;
+
+  int image_pix_size_w = this->width() / 2;
+  int image_pix_size_h = this->height() / 2;
+  imageWindow = new Window(this, "About");                                                                                                         
+  imageWindow->setPosition(Vector2i(0, 0));                                                                                                                   
+  imageWindow->setLayout(new GroupLayout(0,0,0));                                                                                                                     
+  GLTexture t;    
+  t.load(about_png,about_png_size);  
+  float scale = (float)image_pix_size_w / t.width();
+  auto imageView = new ImageView(imageWindow,t);  
+  imageView->setScale(scale);
+  imageView->setFixedScale(true);
+  imageView->setFixedOffset(true);
+  imageView->setFixedWidth(image_pix_size_w);
+  imageView->setFixedHeight(image_pix_size_h);
+  imageWindow->center();
+  //imageWindow->setModal(true);
+
   std::string title = "Yaba Sanshiro "+ std::string(YAB_VERSION) +" Menu";
         window = new Window(this, title);
         window->setPosition(Vector2i(0, 0));
@@ -189,7 +207,7 @@ MenuScreen::MenuScreen( SDL_Window* pwindow, int rwidth, int rheight, const std:
           event.user.data2 = 0;
           SDL_PushEvent(&event);          
         });        
-
+#if 0
         Button *b4 = new Button(tools, "About");
         b4->setFixedWidth(248);
         b4->setCallback([this,b4]() { 
@@ -225,7 +243,7 @@ MenuScreen::MenuScreen( SDL_Window* pwindow, int rwidth, int rheight, const std:
           pushActiveMenu(imageWindow,b4);
 
         });
-
+#endif
         player_configs_[0].player->focusEvent(true);
         player_configs_[0].player->mouseEnterEvent(player_configs_[0].player->absolutePosition(),true);
         mFocus = player_configs_[0].player;

From 1b61637b2da1f52155560f2c347c38f79fd3351a Mon Sep 17 00:00:00 2001
From: devMiyax <smiyaxdev@gmail.com>
Date: Sun, 17 Nov 2019 08:16:24 +0900
Subject: [PATCH 04/12] Add offscreen window for frame buffer reading from cpu

(cherry picked from commit c4efe44db6736d649c8bc114ff66ec40c91fb0c4)
---
 yabause/src/retro_arena/main.cpp | 52 +++++++++++++++++++++++++++-----
 1 file changed, 45 insertions(+), 7 deletions(-)

diff --git a/yabause/src/retro_arena/main.cpp b/yabause/src/retro_arena/main.cpp
index eba29bf38..d1b0c3bbb 100644
--- a/yabause/src/retro_arena/main.cpp
+++ b/yabause/src/retro_arena/main.cpp
@@ -146,7 +146,9 @@ OSD_struct *OSDCoreList[] = {
 
 
 static SDL_Window* wnd;
+static SDL_Window* subwnd;
 static SDL_GLContext glc;
+static SDL_GLContext subglc;
 int g_EnagleFPS = 0;
 int g_resolution_mode = 0;
 int g_keep_aspect_rate = 0;
@@ -183,18 +185,38 @@ void YuiSwapBuffers(void)
 
 int YuiRevokeOGLOnThisThread(){
   LOG("revoke thread\n");
-  SDL_GL_MakeCurrent(wnd,nullptr);
+  int rc = -1;
+  int retry=0;
+  while( rc != 0 ){
+    usleep(16666);
+    retry++;
+    rc = SDL_GL_MakeCurrent(subwnd,subglc);
+    if( rc != 0 ){ LOG("fail revoke thread\n"); }
+    if( retry > 100){
+      LOG("out of retry cont\n");
+      abort();
+    }
+  }
   return 0;
 }
 
 int YuiUseOGLOnThisThread(){
   LOG("use thread\n");
-  SDL_GL_MakeCurrent(wnd,glc);
+  int rc = -1;
+  int retry=0;
+  while( rc != 0 ){
+    usleep(16666);
+    retry++;
+    rc = SDL_GL_MakeCurrent(wnd,glc);
+    if( rc != 0 ){ LOG("fail revoke thread\n"); }
+    if( retry > 100){
+      LOG("out of retry cont\n");
+      abort();
+    }
+  }
   return 0;
 }
 
-}
-
 int saveScreenshot( const char * filename );
 
 int padmode = 0;
@@ -350,12 +372,17 @@ int main(int argc, char** argv)
   SDL_GL_SetAttribute(SDL_GL_DOUBLEBUFFER, 1);
   SDL_GL_SetAttribute(SDL_GL_DEPTH_SIZE, 24);
   SDL_GL_SetAttribute(SDL_GL_STENCIL_SIZE, 8);
+  SDL_GL_SetAttribute(SDL_GL_SHARE_WITH_CURRENT_CONTEXT, 1); 
 
 #if defined(__PC__)
   int width = 1280;
   int height = 720;
   wnd = SDL_CreateWindow("Yaba Snashiro", SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED,
       width, height, SDL_WINDOW_OPENGL | SDL_WINDOW_SHOWN );
+
+  subwnd = SDL_CreateWindow("Yaba Snashiro sub", SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED,
+      width, height, SDL_WINDOW_OPENGL | SDL_WINDOW_HIDDEN );
+
 #else
   int width = dsp.w;
   int height = dsp.h;
@@ -365,11 +392,18 @@ int main(int argc, char** argv)
     printf("Fail to SDL_CreateWindow Bye! (%s)", SDL_GetError() );
     return -1;
   }
+  subwnd = SDL_CreateWindow("Yaba Snashiro sub", SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED,
+      width, height, SDL_WINDOW_OPENGL | SDL_WINDOW_HIDDEN );
 #endif
   dsp.refresh_rate = 60;
   SDL_SetWindowDisplayMode(wnd,&dsp);
   SDL_GetWindowSize(wnd,&width,&height);
   SDL_SetWindowInputFocus(wnd);
+  subglc = SDL_GL_CreateContext(wnd);
+  if(subglc == nullptr ) {
+    printf("Fail to SDL_GL_CreateContext Bye! (%s)", SDL_GetError() );
+    return -1;
+  }
   glc = SDL_GL_CreateContext(wnd);
   if(glc == nullptr ) {
     printf("Fail to SDL_GL_CreateContext Bye! (%s)", SDL_GetError() );
@@ -395,6 +429,7 @@ int main(int argc, char** argv)
   VIDCore->Resize(0,0,width,height,1,p->getInt("Aspect rate",0));
   delete p;
   
+  
   SDL_GL_MakeCurrent(wnd,nullptr);
 #if defined(__RP64__) || defined(__N2__)
   YabThreadSetCurrentThreadAffinityMask(0x4);
@@ -466,14 +501,15 @@ int main(int argc, char** argv)
         }else{
           menu_show = true;
           ScspMuteAudio(1);
+          SDL_GL_MakeCurrent(subwnd,nullptr);
           VdpRevoke();
+          YuiUseOGLOnThisThread();
 
           char pngname_base[256];
           snprintf(pngname_base,256,"%s/%s_", s_savepath, cdip->itemnum);
           menu->setCurrentGameId(pngname_base);
 
           inputmng->setMenuLayer(menu);
-          SDL_GL_MakeCurrent(wnd,glc);
           saveScreenshot(tmpfilename.c_str());
           glUseProgram(0);
           glGetError();
@@ -616,10 +652,11 @@ void hideMenuScreen(){
   glClearColor(0.0f, 0.0f, 0.0f, 1);
   glClear(GL_COLOR_BUFFER_BIT);
   SDL_GL_SwapWindow(wnd);          
-  glClear(GL_COLOR_BUFFER_BIT);
-  SDL_GL_SwapWindow(wnd);          
+  //glClear(GL_COLOR_BUFFER_BIT);
+  //SDL_GL_SwapWindow(wnd);          
   SDL_GL_MakeCurrent(wnd,nullptr);
   VdpResume();
+  YuiRevokeOGLOnThisThread();
   ScspUnMuteAudio(1); 
 }
 
@@ -753,3 +790,4 @@ int saveScreenshot( const char * filename ){
     if(row_pointers) free(row_pointers);
     return rtn;
 }
+}
\ No newline at end of file

From d5f342d831261b93897db6a4ddda63dde06f8b12 Mon Sep 17 00:00:00 2001
From: devMiyax <smiyaxdev@gmail.com>
Date: Thu, 28 Nov 2019 23:53:42 +0900
Subject: [PATCH 05/12] revert frame limit

(cherry picked from commit f970643dbd3b939057be97b90d6ae0034705d336)
---
 yabause/src/vdp2.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/yabause/src/vdp2.c b/yabause/src/vdp2.c
index 5bf9bbc3b..bb6057435 100644
--- a/yabause/src/vdp2.c
+++ b/yabause/src/vdp2.c
@@ -1052,7 +1052,6 @@ void vdp2VBlankOUT(void) {
          framestoskip = 1;
       }else if ((onesecondticks+diffticks) < ((yabsys.OneFrameTime * (u64)framecount) - (yabsys.OneFrameTime / 2)))
       {
-#if 0
          // Check to see if we need to limit speed at all
          for (;;)
          {
@@ -1061,7 +1060,6 @@ void vdp2VBlankOUT(void) {
             if ((onesecondticks+diffticks) >= (yabsys.OneFrameTime * (u64)framecount))
                break;
          }
-#endif
       }
 
       onesecondticks += diffticks;

From ed258e941ceaf2c68bdeac20bdbeb509f1f95607 Mon Sep 17 00:00:00 2001
From: devMiyax <smiyaxdev@gmail.com>
Date: Thu, 28 Nov 2019 23:55:34 +0900
Subject: [PATCH 06/12] keep aspect ratio of on-screen controller

(cherry picked from commit bb1100e7f2d0807e552fbe964541a49cd5e5500d)
---
 .../app/src/main/java/org/uoyabause/android/YabausePad.java   | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/yabause/src/android/app/src/main/java/org/uoyabause/android/YabausePad.java b/yabause/src/android/app/src/main/java/org/uoyabause/android/YabausePad.java
index c3bfdb2d5..37eaacf81 100644
--- a/yabause/src/android/app/src/main/java/org/uoyabause/android/YabausePad.java
+++ b/yabause/src/android/app/src/main/java/org/uoyabause/android/YabausePad.java
@@ -679,10 +679,10 @@ void setPadScale( int width, int height ){
 
         if (width > height) {
             wscale = (float) width / basewidth;
-            hscale = (float) height / baseheight;
+            hscale = wscale; //(float) height / baseheight;
         }else{
             wscale = (float) width / baseheight;
-            hscale = (float) height / basewidth;
+            hscale = wscale; //(float) height / basewidth;
         }
 
         int bitmap_height = bitmap_pad_right.getHeight();

From 5d232d91f041dee539faab8b2061e6a8b3710e49 Mon Sep 17 00:00:00 2001
From: devMiyax <smiyaxdev@gmail.com>
Date: Thu, 28 Nov 2019 23:55:58 +0900
Subject: [PATCH 07/12] v2.10.3

(cherry picked from commit 7e2abc2b2b28e922b32eb9c0b87f8b8b85630e1b)
---
 yabause/CMakeLists.txt               | 2 +-
 yabause/src/android/app/build.gradle | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/yabause/CMakeLists.txt b/yabause/CMakeLists.txt
index 59343700b..a7f034a98 100644
--- a/yabause/CMakeLists.txt
+++ b/yabause/CMakeLists.txt
@@ -19,7 +19,7 @@ endif(YAB_FORCE_GLES20)
 set(YAB_PACKAGE yabause)
 set(YAB_VERSION_MAJOR 2)
 set(YAB_VERSION_MINOR 10)
-set(YAB_VERSION_PATCH 2)
+set(YAB_VERSION_PATCH 3)
 set(YAB_VERSION "${YAB_VERSION_MAJOR}.${YAB_VERSION_MINOR}.${YAB_VERSION_PATCH}")
 
 include(GetGitRevisionDescription)
diff --git a/yabause/src/android/app/build.gradle b/yabause/src/android/app/build.gradle
index 03abf0cf2..67ca9e8a5 100644
--- a/yabause/src/android/app/build.gradle
+++ b/yabause/src/android/app/build.gradle
@@ -45,8 +45,8 @@ android {
         minSdkVersion 21
         targetSdkVersion 28
         multiDexEnabled true
-        versionCode 101
-        versionName '2.10.2'
+        versionCode 102
+        versionName '2.10.3'
         multiDexEnabled true
         vectorDrawables.useSupportLibrary = true
         signingConfig signingConfigs.pro

From 198b5659dbf9d9dfd34399d2ce13138c2ebaecd9 Mon Sep 17 00:00:00 2001
From: devMiyax <smiyaxdev@gmail.com>
Date: Sat, 30 Nov 2019 01:34:41 +0000
Subject: [PATCH 08/12] Sync All Textures

(cherry picked from commit 94b5e79d5fb8a162b104afc348a85aec91be1101)
---
 yabause/src/ygles.c | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/yabause/src/ygles.c b/yabause/src/ygles.c
index 88bad6749..01da6a685 100644
--- a/yabause/src/ygles.c
+++ b/yabause/src/ygles.c
@@ -520,7 +520,7 @@ YglTextureManager * YglTMInit(unsigned int w, unsigned int h) {
   
     glBindTexture(GL_TEXTURE_2D, tm->textureID_in[i]);
     glBindBuffer(GL_PIXEL_UNPACK_BUFFER, tm->pixelBufferID_in[i]);
-    tm->texture_in[i] = (unsigned int *)glMapBufferRange(GL_PIXEL_UNPACK_BUFFER, 0, tm->width * tm->height * 4, GL_MAP_WRITE_BIT | GL_MAP_UNSYNCHRONIZED_BIT);
+    tm->texture_in[i] = (unsigned int *)glMapBufferRange(GL_PIXEL_UNPACK_BUFFER, 0, tm->width * tm->height * 4, GL_MAP_WRITE_BIT );
     if ((error = glGetError()) != GL_NO_ERROR)
     {
       YGLDEBUG("Fail to init YglTM->texture %04X", error);
@@ -597,10 +597,10 @@ void YglTmPull(YglTextureManager * tm, u32 flg){
     glBindBuffer(GL_PIXEL_UNPACK_BUFFER, tm->pixelBufferID_in[tm->current]);
 
     if (flg) {
-      tm->texture_in[tm->current] = (int*)glMapBufferRange(GL_PIXEL_UNPACK_BUFFER, 0, tm->width * tm->height * 4, GL_MAP_WRITE_BIT /*| GL_MAP_INVALIDATE_BUFFER_BIT*/ | GL_MAP_UNSYNCHRONIZED_BIT);
+      tm->texture_in[tm->current] = (int*)glMapBufferRange(GL_PIXEL_UNPACK_BUFFER, 0, tm->width * tm->height * 4, GL_MAP_WRITE_BIT /*| GL_MAP_INVALIDATE_BUFFER_BIT*/ );
     }
     else {
-      tm->texture_in[tm->current] = (int*)glMapBufferRange(GL_PIXEL_UNPACK_BUFFER, 0, tm->width * tm->height * 4, GL_MAP_WRITE_BIT | GL_MAP_INVALIDATE_BUFFER_BIT | GL_MAP_UNSYNCHRONIZED_BIT);
+      tm->texture_in[tm->current] = (int*)glMapBufferRange(GL_PIXEL_UNPACK_BUFFER, 0, tm->width * tm->height * 4, GL_MAP_WRITE_BIT | GL_MAP_INVALIDATE_BUFFER_BIT);
     }
     if (tm->texture_in[tm->current] == NULL) {
       abort();
@@ -677,7 +677,7 @@ void YglTMRealloc(YglTextureManager * tm, unsigned int width, unsigned int heigh
   for (int i = 0; i < NUM_TEXTURE_BUFFER; i++) {
 
     glBindBuffer(GL_PIXEL_UNPACK_BUFFER, new_pixelBufferID[i]);
-    new_texture[i] = (unsigned int *)glMapBufferRange(GL_PIXEL_UNPACK_BUFFER, 0, width * height * 4, GL_MAP_WRITE_BIT | GL_MAP_UNSYNCHRONIZED_BIT);
+    new_texture[i] = (unsigned int *)glMapBufferRange(GL_PIXEL_UNPACK_BUFFER, 0, width * height * 4, GL_MAP_WRITE_BIT );
     if ((error = glGetError()) != GL_NO_ERROR) {
       YGLDEBUG("Fail to init new_texture %04X", error);
       abort();
@@ -4385,7 +4385,7 @@ u32 * YglGetLineColorPointer(){
 
   glBindTexture(GL_TEXTURE_2D, _Ygl->lincolor_tex);
   glBindBuffer(GL_PIXEL_UNPACK_BUFFER, _Ygl->linecolor_pbo);
-  _Ygl->lincolor_buf = (u32 *)glMapBufferRange(GL_PIXEL_UNPACK_BUFFER, 0, 512 * 4, GL_MAP_WRITE_BIT | GL_MAP_INVALIDATE_BUFFER_BIT | GL_MAP_UNSYNCHRONIZED_BIT );
+  _Ygl->lincolor_buf = (u32 *)glMapBufferRange(GL_PIXEL_UNPACK_BUFFER, 0, 512 * 4, GL_MAP_WRITE_BIT | GL_MAP_INVALIDATE_BUFFER_BIT  );
   if ((error = glGetError()) != GL_NO_ERROR)
   {
     YGLLOG("Fail to init YglTM->lincolor_buf %04X", error);
@@ -4449,7 +4449,7 @@ u32* YglGetBackColorPointer() {
   if( _Ygl->backcolor_buf != NULL ){
     glUnmapBuffer(GL_PIXEL_UNPACK_BUFFER);
   }
-  _Ygl->backcolor_buf = (u32 *)glMapBufferRange(GL_PIXEL_UNPACK_BUFFER, 0, 512 * 4, GL_MAP_WRITE_BIT | GL_MAP_INVALIDATE_BUFFER_BIT | GL_MAP_UNSYNCHRONIZED_BIT );
+  _Ygl->backcolor_buf = (u32 *)glMapBufferRange(GL_PIXEL_UNPACK_BUFFER, 0, 512 * 4, GL_MAP_WRITE_BIT | GL_MAP_INVALIDATE_BUFFER_BIT );
   if ((error = glGetError()) != GL_NO_ERROR)
   {
     YGLLOG("Fail to init YglTM->backcolor_buf %04X", error);
@@ -4637,7 +4637,7 @@ u32 * YglGetPerlineBuf(YglPerLineInfo * perline, int linecount, int depth ){
 
   glBindTexture(GL_TEXTURE_2D, perline->lincolor_tex);
   glBindBuffer(GL_PIXEL_UNPACK_BUFFER, perline->linecolor_pbo);
-  perline->lincolor_buf = (u32 *)glMapBufferRange(GL_PIXEL_UNPACK_BUFFER, 0, linecount * 4 * depth, GL_MAP_WRITE_BIT | GL_MAP_INVALIDATE_BUFFER_BIT | GL_MAP_UNSYNCHRONIZED_BIT );
+  perline->lincolor_buf = (u32 *)glMapBufferRange(GL_PIXEL_UNPACK_BUFFER, 0, linecount * 4 * depth, GL_MAP_WRITE_BIT | GL_MAP_INVALIDATE_BUFFER_BIT  );
   if ((error = glGetError()) != GL_NO_ERROR)
   {
     YGLLOG("Fail to init YglTM->lincolor_buf %04X", error);

From ce3331d3906d9726205b647d33001fdef5c10202 Mon Sep 17 00:00:00 2001
From: devMiyax <smiyaxdev@gmail.com>
Date: Sat, 30 Nov 2019 01:33:53 +0000
Subject: [PATCH 09/12] Fix: SEGA rally  does not enter menu sceine

(cherry picked from commit 7b9a0cdb885cee498d5f10d5c0229c287dea3ad9)
---
 yabause/src/vdp2.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/yabause/src/vdp2.c b/yabause/src/vdp2.c
index bb6057435..7fb5e6423 100644
--- a/yabause/src/vdp2.c
+++ b/yabause/src/vdp2.c
@@ -706,8 +706,8 @@ void Vdp2HBlankOUT(void) {
       YabClearEventQueue(vdp1_rcv_evqueue);
       FRAMELOG("**WAIT END**");
       FrameProfileAdd("DirectDraw sync");
+      yabsys.wait_line_count = -1;
       if (Vdp1External.status == VDP1_STATUS_IDLE) {
-        yabsys.wait_line_count = -1;
         Vdp1Regs->EDSR |= 2;
         Vdp1Regs->COPR = Vdp1Regs->addr >> 3;
         ScuSendDrawEnd();
@@ -872,6 +872,7 @@ void vdp2VBlankOUT(void) {
   static u64 onesecondticks = 0;
   static VideoInterface_struct * saved = NULL;
   int isrender = 0;
+  u64 starttime = YabauseGetTicks();
   VdpLockVram();
   FRAMELOG("***** VOUT(T) swap=%d,plot=%d*****", Vdp1External.swap_frame_buffer, Vdp1External.frame_change_plot);
 
@@ -1066,6 +1067,15 @@ void vdp2VBlankOUT(void) {
       lastticks = curticks;
    }
    VdpUnLockVram();
+   static FILE * framefp = NULL;
+   if( framefp == NULL ){
+     framefp = fopen("frame.csv","w");
+   }
+
+   if( framefp != NULL ){
+      fprintf(framefp,"%d,%d\n", g_frame_count, (u32)(YabauseGetTicks()-starttime) );
+      fflush(framefp);
+   }
 }
 
 //////////////////////////////////////////////////////////////////////////////

From 7063eb517192089275dc8648fa57c9e329270de5 Mon Sep 17 00:00:00 2001
From: devMiyax <smiyaxdev@gmail.com>
Date: Sat, 30 Nov 2019 17:09:03 +0900
Subject: [PATCH 10/12] Make PROFILE_RENDERING selective

(cherry picked from commit 9afcfdbab5fb63a7164629c1a2cadee3dfd181a5)
---
 yabause/src/vdp2.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/yabause/src/vdp2.c b/yabause/src/vdp2.c
index 7fb5e6423..77613222c 100644
--- a/yabause/src/vdp2.c
+++ b/yabause/src/vdp2.c
@@ -96,6 +96,7 @@ YabMutex * vrammutex = NULL;
 int g_frame_count = 0;
 
 //#define LOG yprintf
+#define PROFILE_RENDERING 0
 
 YabEventQueue * command_ = NULL;;
 
@@ -872,7 +873,9 @@ void vdp2VBlankOUT(void) {
   static u64 onesecondticks = 0;
   static VideoInterface_struct * saved = NULL;
   int isrender = 0;
+#if PROFILE_RENDERING
   u64 starttime = YabauseGetTicks();
+#endif
   VdpLockVram();
   FRAMELOG("***** VOUT(T) swap=%d,plot=%d*****", Vdp1External.swap_frame_buffer, Vdp1External.frame_change_plot);
 
@@ -1067,15 +1070,21 @@ void vdp2VBlankOUT(void) {
       lastticks = curticks;
    }
    VdpUnLockVram();
+#if PROFILE_RENDERING
    static FILE * framefp = NULL;
    if( framefp == NULL ){
+#if defined(ANDROID)
+     framefp = fopen("/mnt/sdcard/frame.csv", "w");
+#else
      framefp = fopen("frame.csv","w");
+#endif
    }
 
    if( framefp != NULL ){
       fprintf(framefp,"%d,%d\n", g_frame_count, (u32)(YabauseGetTicks()-starttime) );
       fflush(framefp);
    }
+#endif
 }
 
 //////////////////////////////////////////////////////////////////////////////

From 990d46540915927d60dfb8d201d33f3b44d25d62 Mon Sep 17 00:00:00 2001
From: devMiyax <smiyaxdev@gmail.com>
Date: Sat, 30 Nov 2019 18:19:06 +0900
Subject: [PATCH 11/12] v.2.10.4

(cherry picked from commit 058733fcaa712301bb5a7beee6e2b3313e0a1361)
---
 yabause/CMakeLists.txt               | 2 +-
 yabause/src/android/app/build.gradle | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/yabause/CMakeLists.txt b/yabause/CMakeLists.txt
index a7f034a98..12c3cd1f2 100644
--- a/yabause/CMakeLists.txt
+++ b/yabause/CMakeLists.txt
@@ -19,7 +19,7 @@ endif(YAB_FORCE_GLES20)
 set(YAB_PACKAGE yabause)
 set(YAB_VERSION_MAJOR 2)
 set(YAB_VERSION_MINOR 10)
-set(YAB_VERSION_PATCH 3)
+set(YAB_VERSION_PATCH 4)
 set(YAB_VERSION "${YAB_VERSION_MAJOR}.${YAB_VERSION_MINOR}.${YAB_VERSION_PATCH}")
 
 include(GetGitRevisionDescription)
diff --git a/yabause/src/android/app/build.gradle b/yabause/src/android/app/build.gradle
index 67ca9e8a5..3636302be 100644
--- a/yabause/src/android/app/build.gradle
+++ b/yabause/src/android/app/build.gradle
@@ -45,8 +45,8 @@ android {
         minSdkVersion 21
         targetSdkVersion 28
         multiDexEnabled true
-        versionCode 102
-        versionName '2.10.3'
+        versionCode 103
+        versionName '2.10.4'
         multiDexEnabled true
         vectorDrawables.useSupportLibrary = true
         signingConfig signingConfigs.pro

From 933d4e33d3a6382bca270b73e7b27d3ad82727b0 Mon Sep 17 00:00:00 2001
From: devMiyax <smiyaxdev@gmail.com>
Date: Sun, 1 Dec 2019 07:47:43 +0900
Subject: [PATCH 12/12] Revert "Add offscreen window for frame buffer reading
 from cpu"

This reverts commit 1b61637b2da1f52155560f2c347c38f79fd3351a.
---
 yabause/src/retro_arena/main.cpp | 52 +++++---------------------------
 1 file changed, 7 insertions(+), 45 deletions(-)

diff --git a/yabause/src/retro_arena/main.cpp b/yabause/src/retro_arena/main.cpp
index d1b0c3bbb..eba29bf38 100644
--- a/yabause/src/retro_arena/main.cpp
+++ b/yabause/src/retro_arena/main.cpp
@@ -146,9 +146,7 @@ OSD_struct *OSDCoreList[] = {
 
 
 static SDL_Window* wnd;
-static SDL_Window* subwnd;
 static SDL_GLContext glc;
-static SDL_GLContext subglc;
 int g_EnagleFPS = 0;
 int g_resolution_mode = 0;
 int g_keep_aspect_rate = 0;
@@ -185,38 +183,18 @@ void YuiSwapBuffers(void)
 
 int YuiRevokeOGLOnThisThread(){
   LOG("revoke thread\n");
-  int rc = -1;
-  int retry=0;
-  while( rc != 0 ){
-    usleep(16666);
-    retry++;
-    rc = SDL_GL_MakeCurrent(subwnd,subglc);
-    if( rc != 0 ){ LOG("fail revoke thread\n"); }
-    if( retry > 100){
-      LOG("out of retry cont\n");
-      abort();
-    }
-  }
+  SDL_GL_MakeCurrent(wnd,nullptr);
   return 0;
 }
 
 int YuiUseOGLOnThisThread(){
   LOG("use thread\n");
-  int rc = -1;
-  int retry=0;
-  while( rc != 0 ){
-    usleep(16666);
-    retry++;
-    rc = SDL_GL_MakeCurrent(wnd,glc);
-    if( rc != 0 ){ LOG("fail revoke thread\n"); }
-    if( retry > 100){
-      LOG("out of retry cont\n");
-      abort();
-    }
-  }
+  SDL_GL_MakeCurrent(wnd,glc);
   return 0;
 }
 
+}
+
 int saveScreenshot( const char * filename );
 
 int padmode = 0;
@@ -372,17 +350,12 @@ int main(int argc, char** argv)
   SDL_GL_SetAttribute(SDL_GL_DOUBLEBUFFER, 1);
   SDL_GL_SetAttribute(SDL_GL_DEPTH_SIZE, 24);
   SDL_GL_SetAttribute(SDL_GL_STENCIL_SIZE, 8);
-  SDL_GL_SetAttribute(SDL_GL_SHARE_WITH_CURRENT_CONTEXT, 1); 
 
 #if defined(__PC__)
   int width = 1280;
   int height = 720;
   wnd = SDL_CreateWindow("Yaba Snashiro", SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED,
       width, height, SDL_WINDOW_OPENGL | SDL_WINDOW_SHOWN );
-
-  subwnd = SDL_CreateWindow("Yaba Snashiro sub", SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED,
-      width, height, SDL_WINDOW_OPENGL | SDL_WINDOW_HIDDEN );
-
 #else
   int width = dsp.w;
   int height = dsp.h;
@@ -392,18 +365,11 @@ int main(int argc, char** argv)
     printf("Fail to SDL_CreateWindow Bye! (%s)", SDL_GetError() );
     return -1;
   }
-  subwnd = SDL_CreateWindow("Yaba Snashiro sub", SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED,
-      width, height, SDL_WINDOW_OPENGL | SDL_WINDOW_HIDDEN );
 #endif
   dsp.refresh_rate = 60;
   SDL_SetWindowDisplayMode(wnd,&dsp);
   SDL_GetWindowSize(wnd,&width,&height);
   SDL_SetWindowInputFocus(wnd);
-  subglc = SDL_GL_CreateContext(wnd);
-  if(subglc == nullptr ) {
-    printf("Fail to SDL_GL_CreateContext Bye! (%s)", SDL_GetError() );
-    return -1;
-  }
   glc = SDL_GL_CreateContext(wnd);
   if(glc == nullptr ) {
     printf("Fail to SDL_GL_CreateContext Bye! (%s)", SDL_GetError() );
@@ -429,7 +395,6 @@ int main(int argc, char** argv)
   VIDCore->Resize(0,0,width,height,1,p->getInt("Aspect rate",0));
   delete p;
   
-  
   SDL_GL_MakeCurrent(wnd,nullptr);
 #if defined(__RP64__) || defined(__N2__)
   YabThreadSetCurrentThreadAffinityMask(0x4);
@@ -501,15 +466,14 @@ int main(int argc, char** argv)
         }else{
           menu_show = true;
           ScspMuteAudio(1);
-          SDL_GL_MakeCurrent(subwnd,nullptr);
           VdpRevoke();
-          YuiUseOGLOnThisThread();
 
           char pngname_base[256];
           snprintf(pngname_base,256,"%s/%s_", s_savepath, cdip->itemnum);
           menu->setCurrentGameId(pngname_base);
 
           inputmng->setMenuLayer(menu);
+          SDL_GL_MakeCurrent(wnd,glc);
           saveScreenshot(tmpfilename.c_str());
           glUseProgram(0);
           glGetError();
@@ -652,11 +616,10 @@ void hideMenuScreen(){
   glClearColor(0.0f, 0.0f, 0.0f, 1);
   glClear(GL_COLOR_BUFFER_BIT);
   SDL_GL_SwapWindow(wnd);          
-  //glClear(GL_COLOR_BUFFER_BIT);
-  //SDL_GL_SwapWindow(wnd);          
+  glClear(GL_COLOR_BUFFER_BIT);
+  SDL_GL_SwapWindow(wnd);          
   SDL_GL_MakeCurrent(wnd,nullptr);
   VdpResume();
-  YuiRevokeOGLOnThisThread();
   ScspUnMuteAudio(1); 
 }
 
@@ -790,4 +753,3 @@ int saveScreenshot( const char * filename ){
     if(row_pointers) free(row_pointers);
     return rtn;
 }
-}
\ No newline at end of file
